Figure S1: Full reaction network of the predator-prey system.
Gray and black parts in the strand sequences emphasize the building of the predator palindrome, not sequence domains. The red star represents the template-bound fluorescent dye and the modlulation of its intensity depending on the hybridization state (single strand versus double strand) of the corresponding extremity of the template. Reaction (11) does not participate in the dynamics of the system, but is included to explain the contribution of the predator to the prey fluorescence.


Figure S2: Experimental limit-cycle for two-colors $\mathrm{PP}_1$ system with different initial conditions. Experiment done in standard conditions (as Fig. 2 of the main text). Up: $G_1=120 n M$; bottom: $G_1=140 n M$; Left: $N_{1, t=0}=10 n M, P_{1, t=0}=30 n M$, right: $N_{1, t=0}=30 n M, P_{1, t=0}=90 n M$. All the trajectories show the experimental evolution of the system from $t=0$ to $t=1600 \mathrm{~min}$. Independently of the initial conditions, the system evolve towards the same stable orbits, which, for given buffer and enzymes conditions, are imposed solely by the template concentration.


1 Methods
In the following discussion, italicized $\mathrm{X}(X)$ denotes the concentration of specie X (e.g., $P_1$ is the concentration of predator $\mathrm{P}_1$ ). "Bst" or "Polymerase" refer to Bst polymerase large fragment and "nicking enzyme" to Nb.BsmI (another nicking enzyme, Nt.BstNBI, is used solely in Section S4-Prey amplification for analytical purposes). PredatorPrey will be abbreviated PP ( $\mathrm{PP}_1$ for the $\mathrm{N}_1, \mathrm{P}_1, \mathrm{G}_1$ system and $\mathrm{PP}_2$ for the $\mathrm{N}_2, \mathrm{P}_2, \mathrm{G}_2$ system).

Buffer and chemical
Bst polymerase and the nicking enzymes were obtained from New England Biolabs (NEB). A working dilution of Bst was prepared by dissolving the original stock solution 5 times in diluent A (NEB) complemented with $0.1 \%$ Triton-X100 and stored at $-20^{\circ} \mathrm{C}$ for further use. Bst was used at $1.87 \%$ of this diluted stock, i.e. $3 \mathrm{unit} / \mathrm{mL}$ or 3.7 nM in the final mix, unless otherwise specified.
$\mathrm{Nb} . \mathrm{BsmI}$ was used at $6 \%$ of the stock solution in the reaction mix (i.e. $600 \mathrm{units} / \mathrm{mL}$ ). This high concentration reflects the facts that nicking enzymes in general are relatively slow catalysts[1]. Here we tried to increase the rate of the nicking reaction as much as possible to simplify the kinetic analysis (see Section S3). In the present case, we did not observe the previously reported inhibiting effect of high concentration of nicking enzymes on polymerase activity[1], presumably because the nicked products are unstable (see Table S2) and melt away very quickly, releasing the enzyme from a potential product inhibited form.

For the degradation reaction we used $t t$ RecJ, the thermophilic equivalent of E. coli's RecJ enzyme from Thermus Thermophilus $[2,3]$, which is a $5^{\prime}->3^{\prime}$ processive exonuclease. The purified enzyme was a kind gift from Ryoji Masui. A working dilution at $2.5 \mu M$ was formed in diluent D (NEB) and stored at $-20^{\circ} \mathrm{C}$. This enzyme was used at 32.5 nM for single PP pair experiments and 15 nM for multispecies systems. Other concentrations, used for kinetic measurements (Fig. 1 and Section S4) are specified when necessary.

Oligonucleotides were obtained from Biomers, with HPLC purification. To avoid any chemical discrepancy between synthetic oligonucleotides and those obtained in situ from enzymatic activity, the prey sequences $\mathrm{N}_1$ and $\mathrm{N}_2$ were ordered with 5' phosphate modification (in the course of the in vitro reaction, new preys are obtained from the nicking of a longer strand, and hence bear a 5 '-phosphate group, whereas standard, unmodified synthetic oligonucleotides are $5^{\prime}-\mathrm{OH}$ ). While this difference is unlikely to have an influence on the PP oscillating reactions, it may be more important in the case of the quantitative measurements of prey concentrations performed by isothermal re-amplification as described in Section S4-Prey amplification.

Throughout the study, we have used a buffer containing 20 mM Tris- $\mathrm{HCl}, 10 \mathrm{mM}\left(\mathrm{NH}_4\right)_2 \mathrm{SO}_4, 10 \mathrm{mM} \mathrm{KCl}, 50 \mathrm{mM} \mathrm{NaCl}, 8 \mathrm{mM} \mathrm{MgSO}{ }_4, 400 \mu \mathrm{M}$ each dNTP (NEB), $0.1 \%$ Synperonic F108 (Aldrich), $5 n g / \mu L$ extremely thermostable single strand binding protein (ETSSB; NEB), $100 \mu \mathrm{~g} / \mathrm{mL}$ BSA (NEB), $2 \mu M$ Netropsin (Sigma Aldrich), 4 mM DTT, $1 \times$ EvaGreenÂ®, $0.5 \times$ ROX (from a standard $100 \times$ dilution provided by Invitrogen) and a pH of 8.8 . A $4 \times$ stock solution of this buffer was prepared (excluding BSA, DTT and ETSSB, which were added, together with enzymes and oligonucleotides, during reaction assembly) and used afterward.

In the following, "standard conditions" therefore refers to this buffer, $\mathrm{T}=46.5^{\circ} \mathrm{C}$, Bst $3.7 \mathrm{nM}, \mathrm{Nb} . \mathrm{BsmI} 600$ units/mL, ttRecJ 32.5 nM .

The functions of the components included in this buffer are summarized below.
- BSA, DTT and Synperonic F108 (a surfactant) act as stabilizing agents for the enzymes. Their inclusion in the buffer allows us to run experiments for up to a few days in closed tubes without excessive loss of enzymatic activity (whereas for example Nb.BsmI is reported by NEB to be stable for less than 8 hours in the recommended digestion buffer, at the recommended temperature).
- EvaGreen is a DNA intercalating green fluorescent dye that does not inhibit enzymatic activity. It has a marked preference for duplex DNA over single strands oligonucleotides. In a PP oscillator, it allows us to observe nonspecifically the appearance/disappearance of double strand DNA. However we have also used the lower signal generated by single strand binding for some kinetic measurement (digestion rate of $\mathrm{N}_1$ and $\mathrm{N}_2$ ). As other intercalating agents, EvaGreen has an effect on DNA melting temperature. Hence the melting temperatures used in Section S4-Predation were measured in its presence.
- Netropsin is a short peptide with a strong affinity for AT-rich sequences in double-stranded DNA. We use it to prevent the occurrence of parasitic DNA species, which tend to appear stochastically when mixtures of polymerase and restriction enzymes are incubated for extended periods of time[4]. Such species emerging and exponentially developing- in isothermal amplification schemes have the potential to disrupt experimental setups[5] or even to contaminate a whole lab[6] and maximum precautions[7] should be taken to minimize their emergence and spreading. The analysis of such parasites induced by polymerase-nicking enzyme mixes by Tan et al. [4] has shown that they typically contain long AT repeats, interspaced by sequences that serves as target of the restriction enzyme present in the buffer. We observed that netropsin tends to delay the development of such species in a contaminated experiment, while having at most a marginal effect on the targeted oscillatory reaction. As a rationale, it may be tentatively proposed that by binding on AT rich sequences, netropsin locally stabilizes the double helix[8] and inhibits the strand slippage and hairpin-coil transition mechanisms that is necessary for the growth of the parasitic specie[4]. However, we have also noticed that a variety of parasitic reactions with exponential fluorescence signatures can appear after long incubation of PP mixtures (even in the presence of netropsin), some of them even being able to coexist with the oscillatory PP reaction.
- ETSSB is a thermostable single-stranded DNA binding protein with a large binding site (from 35 to 65 bases, depending on salt conditions, in the case of E. Coli SSB[9]). SSB has been reported as having a facilitating effect in nicking enzyme mediated DNA amplification[1]. In our case, we have noticed that ETSSB accelerates the growth rate of the prey, possibly by lowering the melting temperature of the G template hairpin. We also observed that ETSSB has an accelerating role in the growth of $\mathrm{P}_1$ predators, less marked for $\mathrm{P}_2$.

Reaction assembly
Reactions were typically run in parallel in strips of $20 \mu \mathrm{~L}$ PCR white tubes. A master mix was first assembled by mixing all the common components at $4^{\circ} \mathrm{C}$. Concentrations of chemicals that were different from tube to tube (e.g. the concentration of template in the experiment shown in Fig. 3d of the main text, or the initial concentrations of prey and predators in the experiment of Fig. 4e), were assembled in a minimal reasonable volume directly in the PCR tube, after which the master mix containing all other components was added. This procedure insures minimum dispersion in the concentration of the fixed components from tube to tube ${ }^1$. Careful mixing was applied ( 10 s vortex) at each assembly stage. The tubes were then quickly centrifuged and transferred in a Bio-Rad CFX or MiniOpticon thermocycler set at a constant temperature of $46.5^{\circ} \mathrm{C}$ (this temperature was selected somewhat arbitrarily, as the oscillating PP reaction can be observed over a range of at least several degrees, provided the enzymes and template concentrations are adjusted).

Monitoring and analysis
The fluorescence records were obtained from the real-time thermocyclers at interval varying from a few seconds (for kinetic measurements described in Section 4) to several minutes (for the longest experiments, e.g. Fig. 3b or Fig. 4). The fluorescence cross-talks between the different channels (green, yellow and deep red) were removed by the built-in software after calibration for EvaGreen (ex. 500 nm , em. 530 nm ), Dy530 (ex. 539 nm , em. 561 $\mathrm{nm})$, JOE (ex. 520 nm , em. 548 nm ) and Dy681 (ex. 691 nm , em. 708 nm ). A third order polynomial base line was subtracted from these crude data sets to eliminate the background fluorescence and long term drifts whose causes were obviously not relevant to the reactions of interest (physical effects, evaporation, dye bleaching, slow side reactions, etc. ..). In the case of Dy681 the fluorescence ratio Dy681/ROX was used instead because it tended to be more stable.

The green fluorescence produced by EvaGreen contains a contribution from each of the species shown in Fig. S1. Because of the double-strand preference of EvaGreen, the signal is dominated by the contribution of double-stranded species. Free preys are shorter than predators and don't form stable hairpins or duplex, hence they produce less fluorescence. Duplexes including the template (template-prey but also possibly non-productive template-predator) may also contribute to the signal, but will be at least partially quenched in the green channel by the presence of the template-bound dye of higher wavelength. Rather than trying to obtain the fluorescent contribution of each molecular species (which can be dependent on the experimental conditions and optical set-up), we instead define the green signal as a linear combination of the prey and predator concentrations, dominated by the latter. Following a similar argument (but considering only the species involving the template), we define the yellow (or red) signal as a linear combination dominated by the prey concentration and containing a predator contribution (see reaction (11) in Fig. S1). Because of the linearity, we directly obtain that predator and prey are also linear combinations of the green and yellow signal. Hence, in arbitrary fluorescent units, we just need to find out the cross-talk of predators in the mostly preys signal (yellow or red) and that of the preys in the mostly predators signal (green). These parameters are determined empirically by minimizing long tails for preys spikes and pre-bumps for predators spikes. This process is applied at once to one set of experiment, for example all the plots of Fig. 3d,e share the same cross-talk parameters (in this specific case, the cross-talk of predators to preys was found at $28 \%$ and that of prey to predators was found at $22 \%$ ). Cross-talks talks for the $P P_2$ pair were typically lower. These corrected signals correspond to the "Green (Predator) shift" and the "Yellow (Prey) shift" (or "Red (Prey) shift") plotted for experiments with a single PP pair and the symbiotic experiments of Fig. 5a. In the case of multiple PP pairs, this procedure cannot be applied since we observe only three signals for four species. Hence in this case we only remove the preys contribution to the $P_1+P_2$ signal (i.e. we don't correct for the small contributions of the predators to their respective prey signals).


${ }^1$ When comparing two experiments that were not created from the same master mix (e.g.- experiments performed on two different days) one should assume some dispersion on the concentrations, as a result of unavoidable uncertainties on pipetted volumes. For reaction performed at an interval of several weeks, a change of enzymatic activity, due to the aging of the enzyme stocks should be further expected.



2 Sequence design
The design of the basic PP reaction network comes down to solely selecting the sequence of the G template, in such a way that the reactivity of the system organizes itself along the reaction network depicted in Fig. S1.

This mapping can be reduced to a short set of design constraints:
D Predators should have a palindromic sequence because of their templated duplication mechanism (reaction 7).
- Moreover, this palindrome should be as short as possible so that predator duplexes significantly dissociate (reaction 8 ) -avoiding product inhibition. While this requirement may seem contradictory with the fact that the prey needs to form a primer-template with the predator for the reaction to proceed (reaction 6 ), we have observed that Bst polymerase is able to extend efficiently such short and unstable partial duplexes, even when working well above their melting temperature.
- In order to limit the length of the predator, a part of the palindromic sequence was already included in the prey's 3' end sequence. If this palindrome is too long however, preys may become able to form so-called primerdimers because they possess an overlapping self-complementary sequence. This may lead to an unwanted leak reaction if such primer-dimers are extended by the polymerase. After some trial-and-errors, we found that a length of 10 bases for the prey, 6 of which are palindromic, and hence a 14 base predator was appropriate for predator growth in the $40-50^{\circ} \mathrm{C}$ temperature range, without leading to a significant amount of the afore mentioned leak reaction (at least for prey concentrations in the hundred of nano-molar range, see also Section S4-Prey amplification).
- The prey should further incorporate the recognition sequence of a nicking enzyme, itself stable and active over the working temperature range (i.e. thermophilic). This is necessary for the prey's extension-nicking replication mechanism (reaction 3 and 4).

D Prey amplification and predator growth reactions are compatible (in the same mix) at the cost of one further constraint: the nicking enzyme recognition site should be such that a single prey sequence does not contain the recognition site (only a dual repeat of prey does). Otherwise this sequence would be transferred to the predator and would render them sensitive to being nicked -back to prey. To satisfy this constraint we selected $\mathrm{Nb} . \mathrm{BsmI}$ as the nicking enzyme for this work. ${ }^2$. It recognizes the double stranded sequence $5^{\prime}-\mathrm{G}^{\wedge}$ CATTC-3' and cuts within this site, at the position marked by a hat $\left({ }^{\wedge}\right)$. Hence this recognition sequence, in double strand form, is found only on G duplexes and nowhere else in the system (reaction 4).
- Finally the template sustaining prey growth should be stable against the exonuclease, but not preys themselves, nor predators (reactions 9 and 10). As already shown[10], this can be achieved with a few phosphorothioates backbone modifications.

The sequences are then decided by combining these constraints. The positions that remained free (unconstrained) in the sequence are assigned random bases and we selected two individual (Table S1) with a prey-template melting temperature (at 100 nM ) slightly below the working temperature (Table S2).

Additional, non-genetic modifications of the template were used to protect it from exonuclease degradation, block its extension by the polymerase and use it as a fluorescent reporter of prey concentration:
- Since ttRecJ is a processive exonuclease digesting from the 5' end of single-strand oligonucleotides, we expected that phosphorothioates backbone modifications on this side of the template should provide a good protection[11]. Indeed templates bearing three modifications in a row were almost insensitive to ttRecJ degradation. We also verified that this modification did not significantly affect polymerase or nicking enzyme activity on the template.
- In the targeted reaction set the $3^{\prime}$ end of the template is never in a position to serve as a primer for the polymerase (and hence to be extended). Still we took the precaution to block this end with a phosphate modification. Since only $3^{\prime} \mathrm{OH}$ ends can be extended by DNA polymerases, this precaution serves to prevent potential leak reactions.
- For templates used as prey reporter, this terminal 3 '-phosphate was replaced by a fluorescent dye. The mechanism by which such modified templates report on their input strand (prey) concentration is detailed elsewhere[12]. It should be noted that these modifications of the template 3' end may not be neutral with respect to the kinetic parameters of the system (it will slightly change the associated melting temperatures[13], but may also affect the recognition by some enzymes). We observed indeed that template 3 ' modifications had an impact on the dynamic of the system. In practice however, we found that they could be compensated for by varying the concentration of template or the reaction conditions.

${ }^2$ The use of this enzyme is recommended at $65^{\circ} \mathrm{C}$ but we found that it retain a correct activity -at least fo the nicking of short
duplexes like the one of interest here- down to $40{ }^{\circ} \mathrm{C}$


The sequence of Sy was chosen at random (after checking the absence of cross-talks with the previously designed species), and contain the nicking enzyme recognition sequence. The sequences of the symbiotic templates $\mathrm{G}_{2 \rightarrow \mathrm{Sy}}$ and $\mathrm{G}_{\mathrm{Sy} \rightarrow 2}$ (as well as $\mathrm{G}_{2 \rightarrow 1}$ and $\mathrm{G}_{1 \rightarrow 2}$ ) were then simply formed by combination of the corresponding input and output sequences.

Mes $\mathrm{N}_1$ and $\mathrm{MesN}_2$ are used in Section S4-Prey amplification to quantitatively measure the concentrations of $\mathrm{N}_1$ and $\mathrm{N}_2$, respectively. These re-amplification templates incorporate the recognition sequence of the nicking enzyme Nt.BstNBI[4].


\begin{table}
\captionsetup{labelformat=empty}
\caption{Table S1: Sequences and modification of the oligonucleotides used in this study.}
\begin{tabular}{|l|l|l|}
\hline Name & Sequence ( $5^{\prime} \rightarrow 3$ ') & 3'-modifications \\
\hline $\mathrm{N}_1$ & CATTCGGCCG & \\
\hline $\mathrm{P}_1$ & CATTCGG-CCGAATG & \\
\hline $\mathrm{G}_1$ & C*G*G*CCGAATG-CGGCCGAATG & phos or Dy530 (yellow fluorescence) \\
\hline $\mathrm{N}_2$ & CATTCCCGGG & \\
\hline $\mathrm{P}_2$ & CATTCCC-GGGAATG & \\
\hline $\mathrm{G}_2$ & C*C*C*GGGAATG-CCCGGGAATG & phos or Dy681 (deep red fluorescence) \\
\hline Sy & CATTCGGGCG & \\
\hline $\mathrm{G}_{1 \rightarrow 2}$ & C*C*C*GGGAATG-CGGCCGAATG & phos \\
\hline $\mathrm{G}_{2 \rightarrow 1}$ & C*G*G*CCGAATG-CCCGGGAATG & phos \\
\hline $\mathrm{G}_{2 \rightarrow \mathrm{Sy}}$ & C*G*C*CGAATG-CCCGGGAATG & phos \\
\hline $\mathrm{G}_{\mathrm{Sy} \rightarrow 2}$ & C*C*C*GGGAATG-CGCCCGAATG & JOE (yellow fluorescence) \\
\hline $\mathrm{MesN}_1$ & CGGCCGAATG-CACTGACTCGGCCGAATG & \\
\hline $\mathrm{MesN}_2$ & CCCGGGAATG-CACTGACTCCCGGGAATG & \\
\hline
\end{tabular}
\end{table}
* indicates a phosphorothioate backbone modification used to block exonuclease digestion.
The area in bold is the recognition sequence of the nicking enzyme Nb.BsmI (or Nt.BstNBI in the case of $\mathrm{MesN}_1$ and $\mathrm{MesN}_2$; see Section S4-Prey amplification kinetic).

\begin{table}
\captionsetup{labelformat=empty}
\caption{Table S2: Melting temperatures for hybridization of the oligonucleotides in $\mathrm{PP}_1$ and $\mathrm{PP}_2$.}
\begin{tabular}{|l|l|l|l|l|l|l|l|}
\hline & $\mathrm{N}_1$ & $\mathrm{P}_1$ & $\mathrm{G}_1$ & $\mathrm{N}_2$ & $\mathrm{P}_2$ & $\mathrm{G}_2$ & Folding \\
\hline $\mathrm{N}_1$ & $28.3^{\circ} \mathrm{C}$ & $43.4^{\circ} \mathrm{C}$ & $43.8^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $0.1^{\circ} \mathrm{C}$ & $0.9^{\circ} \mathrm{C}$ & $<0{ }^{\circ} \mathrm{C}$ \\
\hline $\mathrm{P}_1$ & & $54.0^{\circ} \mathrm{C}$ & $46.9{ }^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $5.0^{\circ} \mathrm{C}$ & $5.9^{\circ} \mathrm{C}$ & $51.1^{\circ} \mathrm{C}$ \\
\hline $\mathrm{G}_1$ & & & $64.4^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $<0{ }^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $91.6^{\circ} \mathrm{C}$ \\
\hline $\mathrm{N}_2$ & & & & $23.4{ }^{\circ} \mathrm{C}$ & $40.7^{\circ} \mathrm{C}$ & $41.2{ }^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ \\
\hline $\mathrm{P}_2$ & & & & & $52.4^{\circ} \mathrm{C}$ & $44.5^{\circ} \mathrm{C}$ & $51.1^{\circ} \mathrm{C}$ \\
\hline $\mathrm{G}_2$ & & & & & & $61.5^{\circ} \mathrm{C}$ & $91.0^{\circ} \mathrm{C}$ \\
\hline Sy & & & $36.5^{\circ} \mathrm{C}$ & & & $15.3^{\circ} \mathrm{C}$ & \\
\hline
\end{tabular}
\end{table}
The area in white shows the intra-pair interactions while the area in gray reports the inter-pair interactions. The blue column describes the folding reactions. Calculated with Dynamelt (http://mfold.rna.albany.edu/?q=DINAMelt) for 100 nM concentrations of each oligos and salt conditions as in experimental setup. Note that the presence of ETSSB and the intercalating dye EvaGreen will affect the actual melting temperatures.



3 A simple two species model obtained from the chemical mechanism
In this section we build a simple, two-variable mathematical model. In the process, we try to maintain as much as possible a chemically reasonable mechanistic description with respect to the chemical reaction network shown in Fig. S1. The goal is in particular to preserve a link between the parameters of this high-level two variables model, and measurable kinetic and thermodynamic parameters of each of the individual, lower-level chemical interactions that compose this predator-prey system.

Simplifications used to reach this model are based on the fact that the polymerase and the exonuclease are processive enzymes, so that even if they perform complicated multistep chemical transformations (e.g. in the case of the polymerase-mediated reaction (3) of Fig. S1, the sequential selection and polymerization of the 10 matching bases necessary to extend a G:N partial duplex to its full length), no intermediate product accumulates and these reactions can be considered as individual chemical events. Another simplifying principle is that, because G and P form hairpins or palindromic duplexes, and considering the relatively high working temperature, their binding to the prey is weak. It thus becomes possible to consider that the amount of duplex species ( $\mathrm{P}: \mathrm{N}$ or $\mathrm{G}: \mathrm{N}$ ) is always a small fraction of the total population of each P and N species.

We denote the concentration of prey as $N$, that of the template as $G$, the concentration of prey-template partial duplex as $G: N$, the concentration of predators as $P$, that of prey-predators partial duplexes $P: N$, the concentration of Bst polymerase as pol and the concentration of ttRecJ enzyme as rec. Other notations are introduced when necessary.

Prey Growth
The prey amplification is a composite process comprising reaction (1) to (5) in Fig. S1.
With an estimated $K_a^{G N}$ of $5 \times 10^{-7} \mathrm{nM}^{-1}$, the association of $\mathrm{N}_1$ and $\mathrm{G}_1$ is very unstable. For example, at 100 nM of each $\mathrm{N}_1$ and $\mathrm{G}_1$, NUPACK (http://www.nupack.org/) predicts less than 15 pM of $\mathrm{N}_1$ : $\mathrm{G}_1$ duplex at the working temperature. We can then assume that the extension of these few partial duplexes by the polymerase will be a first order and rate-limiting step of the amplification process. The same holds for the second predator-prey system ( $\mathrm{PP}_2$ ).

However this extension is followed by a nicking event. This enzymatic mechanism will impose an upper limit on the maximum rate of the process, that we choose to describe empirically with a Michaelis-Menten-like expression. We then write the corresponding flux (i.e. the kinetic expression corresponding to this specific process) as:
$$
\varphi_{N \triangleright N}=k \cdot \operatorname{pol} \frac{G: N}{K+G: N}
$$

Note that the values of $k$ and $K$ should be expected to depend on the nicking enzyme, but because we have kept it fixed throughout this study, we consider these parameters as constants (see Section S6 for discussion on this point).

From reaction (1) in Fig. S1, and considering a fast equilibrium, we have:
$$
G: N=K_a^{G N} \cdot G_{\text {free }} \cdot N_{\text {free }}
$$

The prey species N is involved in a number of equilibria, but all of them have a small binding constant so we can assume that $N_{\text {free }} \approx N_{\text {total }} \approx N$. Similarly, and assuming that the template doesn't accumulate too much in the doubled stranded form, $G_{\text {free }} \approx G_{\text {total }} \approx G$. We can then write:
$$
\varphi_{N \triangleright N}=k \cdot \operatorname{pol} \frac{K_a^{G N} \cdot G \cdot N}{K+K_a^{G N} \cdot G \cdot N}=k_1 \cdot \operatorname{pol} \frac{G \cdot N}{1+b \cdot G \cdot N}
$$
with $k_1=K_a^{G N} \cdot k / K$ and $b=K_a^{G N} / K$.
Predation
This process encompasses reactions (6) to (8) in Fig. S1. We note again that the equilibrium of association of the prey with the predator, (reaction 6), has a low binding constant and is expected to be mostly displaced toward dissociation. At $46.5^{\circ} \mathrm{C}$ and 100 nM of each $\mathrm{P}_1$ and $\mathrm{N}_1$, NUPACK predicts less than 8 nM of P:N duplex. Then, at any time in the reaction, only a small amount of prey-predator duplex forms. If further reaction (8) is fast enough ${ }^3$, the extension to predator duplex should be a first-order and rate-limiting step for the overall process. With these assumptions and the above notations, the duplication of predator, with consumption of one prey give the rise to the flux:
$$
\varphi_{N \triangleright P}=k^{\prime} \cdot \operatorname{pol} \cdot P: N
$$

From reaction (6) in Fig. S1, we have:
$$
P: N=K_a^{P N} \cdot P_{\text {free }} \cdot N_{\text {free }}
$$

As before we have: $N_{\text {free }} \approx N_{\text {total }} \approx N$. We assume a similar relation for the predator. Then, defining $k_2=k^{\prime} \cdot K_a^{P N}$, we obtain:
$$
\varphi_{N \triangleright P}=k^{\prime} \cdot K_a^{P N} \cdot \mathrm{pol} \cdot P \cdot N=k_2 \cdot \mathrm{pol} \cdot P \cdot N
$$

As can be seen in Fig. S9 this simple rate equation is able to describe accurately the predator replication process, at least for concentrations in the hundred of nano-molar range. This supports at posteriori the relevance of the previous approximations. For a more complete treatment of palindromic replicators dynamics, see [14].

Degradation
The exonuclease that we use is a processive enzyme and hence the degradations of both prey and predator correspond to a single-step enzymatic reaction which can be reasonably described by a Michaelis-Menten kinetic expression (eq. 2).
$$
\frac{d X}{d t}=r e c \cdot k_{c a t, X} \frac{X}{K_{m, X}+X}=V_m \frac{X}{K_{m, N}+X}
$$

We note $k_{\text {cat }, N}$ and $k_{\text {cat }, P}$ ttRecJ catalytic rates versus N and P respectively, and $K_{m, N}$ and $K_{m, P}$ the corresponding Michaelis-Menten constants. Because prey and predator use the same enzyme for their degradation, there is a competitive situation, which leads to the two coupled flux equations:
$$
\begin{aligned}
& \varphi_{N \triangleright \varnothing}=\operatorname{rec} \cdot k_{c a t, N} \frac{N}{K_{m, N}\left(1+\frac{N}{K_{m, N}}+\frac{P}{K_{m, P}}\right)} \\
& \varphi_{P \triangleright \varnothing}=\operatorname{rec} \cdot k_{c a t, P} \frac{P}{K_{m, P}\left(1+\frac{N}{K_{m, N}}+\frac{P}{K_{m, P}}\right)}
\end{aligned}
$$

However, considering that $K_{m, N} \gg K_{m, P}$ (see Table S4 in Section S4) and that $\max [N(t)] \approx \max [P(t)] \approx K_{m, P}$ (e.g. Fig. 2 g main text) we can approximate these expressions as:
$$
\begin{aligned}
\varphi_{N \triangleright \varnothing} & =r e c \cdot k_{c a t, N} \frac{N}{K_{m, N}\left(1+\frac{P}{K_{m, P}}\right)} \\
\varphi_{P \triangleright \varnothing} & =r e c \cdot k_{c a t, P} \frac{P}{K_{m, P}\left(1+\frac{P}{K_{m, P}}\right)}
\end{aligned}
$$

Defining the first-order degradation rate for N and P as $k_N=\frac{k_{\text {cat }, N}}{K_{m, N}}$ and $k_P=\frac{k_{\text {cat }, P}}{K_{m, P}}$ we get:
$$
\begin{gathered}
\varphi_{N \triangleright \varnothing}=r e c \cdot k_N \frac{N}{1+\frac{P}{K_{m, P}}} \\
\varphi_{P \triangleright \varnothing}=r e c \cdot k \frac{P}{1+\frac{P}{K_{m, P}}}
\end{gathered}
$$
${ }^3$ ETSSB may also play a role at this stage


Complete system
The kinetic evolution of the prey species is determined by the contribution of growth, predation and degradation reactions, while that of the predator results from predation and degradation. Hence the complete system is dynamically described by the set of two coupled ordinary differential equations (3) and (4).
$$
\begin{gathered}
\dot{N}=\frac{d N}{d t}=k_1 \cdot \mathrm{pol} \cdot G \frac{N}{1+b \cdot G \cdot N}-k_2 \cdot \mathrm{pol} \cdot N \cdot P-\mathrm{rec} \cdot k_N \frac{N}{1+\frac{P}{K_{m, P}}} \\
\dot{P}=\frac{d P}{d t}=k_2 \cdot \mathrm{pol} \cdot N \cdot P-\mathrm{rec} \cdot k_P \frac{P}{1+\frac{P}{K_{m, P}}}
\end{gathered}
$$

We define: $\tau=t / t_c=t \cdot k_2 \cdot \mathrm{pol} \cdot K_{m, P}, n=N / K_{m, P}, p=P / K_{m, P}, g=G / G_0=\frac{k_1 \cdot G}{k_2 \cdot K_{m, P}}, \lambda=k_N / k_P$, $\beta=b \cdot k_2 \cdot K_{m, P}^2 / k_1$, and $\delta=\frac{r e c}{p o l} \cdot \frac{k_P}{k_2 \cdot K_{m, P}}$ and obtain the nondimensionalized equations:
$$
\begin{gathered}
\frac{d n}{d \tau}=\frac{g \cdot n}{1+\beta \cdot g \cdot n}-p \cdot n-\lambda \cdot \delta \frac{n}{1+p} \\
\frac{d p}{d \tau}=p \cdot n-\delta \frac{p}{1+p}
\end{gathered}
$$

Hence, our chemical construction can be modeled with a rather simple version of the predator-prey system of equations. Despite its extreme simplicity, we will see that this model is able to reproduce accurately the experimental observations, for two species systems (Fig. 2 and Figs S13, S14, S15) as well as for more complex multi-species systems (Fig. 3 and 4, Section S6). The relatedness between our model -derived from a simple kinetic description of the chemical reactions- and classic models of theoretical ecology permits a meaningful use of the ecological metaphor to understand the functioning of an abiotic system -and vice versa. Also note that a large body of theoretical work on the molecular origin of life[15] has been built on the "replicator equation"[16] itself analogous[17] to Lotka-Volterra systems.


Bifurcation analysis
Using the optimized values of the kinetic parameters obtained in section S4 and summarized in Table S5 for standard conditions ( $r e c=32.5 n M$ ), we can calculate the non-dimensional parameters and the scaling factors $G_0$ and $t_c$ for standard conditions. They are gathered in Table S3.

\begin{table}
\captionsetup{labelformat=empty}
\caption{Table S3: Nondimensional parameters of the model for standard conditions}
\begin{tabular}{|c|c|c|}
\hline & $\mathrm{PP}_1$ & $\mathrm{PP}_2$ \\
\hline \hline$\beta$ & $8.7 \times 10^{-2}$ & $1.3 \times 10^{-1}$ \\
\hline$\lambda$ & 4.5 & 3.7 \\
\hline$\delta$ & 0.39 & 0.38 \\
\hline$G_0(\mathrm{nM})$ & 53 & 86 \\
\hline$t_c(\mathrm{~min})$ & 2.6 & 3.6 \\
\hline
\end{tabular}
\end{table}

We will now study the behavior of this system as a function of $g$, the scaled template concentration, which can be regarded as the richness of the environment sustaining prey growth, the enzymatic ratio $\delta$, which gives the scaled degradation rate of the predator, and $\lambda$, the relative first-order degradation rate of the prey versus that of the predator. In the following discussion, we study the necessary conditions for the emergence of oscillations in this model.


First-order exonuclease (large $K_{m, P}$ )
In a first step, we study the case where $P$ is very small compared to $K_{m, P}$ at any time, i.e. we put ourselves in the case where the predator never saturates the exonuclease. With this simplification, the stability of the system of eqs. 7 and 8 can be analytically obtained.

Because $p \ll 1$ is always true, the equations become:
$$
\begin{gathered}
\dot{n}=\frac{g \cdot n}{1+\beta \cdot g \cdot n}-p \cdot n-\lambda \cdot \delta \cdot n \\
\dot{p}=p \cdot n-\delta \cdot p
\end{gathered}
$$

We first note that the case of negligible values of $\beta$ (i.e. unrealistic never-saturating production), would bring us to the pure Lotka system, with its pathologic conservative orbits. This classic but physically irrelevant problem has been well studied (see for example[18, 19])

For a realistic system, i.e. with limited grow potential, hence non-negligible values of $\beta$, we show in the following that the system always has a stable fixed point and can not display limit cycles.

By setting $\dot{n}$ and $\dot{p}$ to 0 , one finds three equilibrium points:
$\triangleright(0,0)$ i.e extinction of both species,
$\triangleright\left(\frac{g-\delta \cdot \lambda}{\beta \cdot \delta \cdot \lambda \cdot g}, 0\right)$ i.e. extinction of the predator and stable population of prey, valid for $\lambda \cdot \delta<g$,
$\triangleright\left(\delta, \frac{g}{1+\beta \cdot \delta \cdot g}-\lambda \cdot \delta\right)$, corresponding to coexistence of both species, valid for $g>\lambda \cdot \delta /\left(\beta \cdot \lambda \cdot \delta^2-1\right)$.
Using linear stability analysis (i.e. by looking at the eigenvalues of the community matrix linearized around these points[18]) we find that each of these fixed points is stable for, respectively, $\lambda \cdot \delta>g, \lambda \cdot \delta<g<\lambda \cdot \delta /\left(\beta \cdot \lambda \cdot \delta^2-1\right)$ and $g>\lambda \cdot \delta /\left(\beta \cdot \lambda \cdot \delta^2-1\right)$. Therefore the system can have three behaviors, depending on the relative values of the parameters: for small values of $g$ it evolves toward the $(0,0)$ fixed point (absence of any species). For intermediate values of $g$, it evolves toward a stable state where the steady prey population is not high enough to support a predator population. Finally, for large $g$, predator and prey reach a stable coexistence steady state, possibly after some damped oscillations. It can be seen that this simple model always has a single stable equilibrium for any set of parameters and that no sustained oscillations are permitted (bistability is excluded as well). This system can at most display damped oscillations (see Fig. S3). This behavior (i.e. the nonexistence of sustained oscillations) is a consequence of the saturation of prey growth. Such a stabilizing effect of limited growth potential has already been noticed[19] and the absence of cycle in a similar model with limited prey growth (the Volterra model) been noted[20, 21].

In classical PP models that do display a limit cycle, this stabilizing effect is generally balanced by nonlinearities in the functional response (i.e. the predation term). In our model the functional response is linear. We see below (section S3-General case) that in the present system, it is the saturable decay term that allows stable limit cycles to occur.

In relation with ecological modeling, one may note at this stage that, in the absence of predators, eq. (7) can be rewritten as:
$$
\frac{g \cdot n}{1+\beta \cdot g \cdot n}-\lambda \cdot \delta \frac{n}{1+p}=\frac{(g-\lambda \cdot \delta)}{(1+\beta \cdot g \cdot n)} n\left(1-\frac{\beta \cdot \lambda \cdot \delta \cdot g}{g-\lambda \cdot \delta} n\right)=G R(n) \cdot n \cdot(1-C P \cdot n)
$$

Thus the prey growth in our system has a form close to the logistic equation, one of the most used equation in ecology to describe saturable autocatalytic growth processes. By analogy, the carrying capacity would be $C P= \frac{g-\lambda \cdot \delta}{\beta \cdot \lambda \cdot \delta \cdot g}$, and the growth rate $G R(n)$. $G R(n)$ is given at low prey concentrations $(n \ll 1 /(\beta \cdot g))$ by $g-\lambda \cdot \delta$ and tends to decrease at higher concentration, giving a slightly less pronounced sigmoidal shape to the population growth (see Fig. S4).

Figure S3: Dynamics of the system with first order degradation.
Simulations using eq. (5) and (6). Unspecified parameter values are from Table S3-PP ${ }_1$. (A) Bifurcation diagram in the $g$ - $\lambda$ parameter space, showing the region of existence for the 3 stable equilibria (i.e. species eventually reaching extinction or a steady state noted $X_{s s}$ ). (B) Typical trajectory (dashed line) showing damped oscillations in the $p-n$ space for parameters corresponding to the red spot in $a$ (i.e. in the region of stable coexistence of prey and predator). The prey and predator nullclines are shown in red and blue, respectively. (C) The corresponding time plot, in nondimensional time and concentrations (prey in yellow, predator in green).

Figure S4: Prey growth in a predator-free world. Present chemical model versus logistic equation.
$\tau$ is the nondimensional time and $n$ the nondimensional prey concentration. $n_{t=0}=10^{-3}$. Solid line: present model (eq. 9); dashed line: logistic equation. The parameters used are those given in Table S 3 for $\mathrm{PP}_1$, and $g=2.5$.

Figure S5: Dynamics of the system with saturated degradation.
Model with competitive Michaelis-Menten degradation (eqs. 5 and 6). Unspecified parameters values are from Table S3-PP ${ }_1$.
(A) Bifurcation diagram in the $g-\lambda$ parameters space, showing the region of existence for stable equilibria and oscillations.
(B) Typical trajectory (dashed line) showing sustained oscillations in the $\mathrm{p}-\mathrm{n}$ space for parameters corresponding to the red spot in $a$ (i.e. in the region of stable oscillations). The prey and predator nullclines are shown in red and blue, respectively. (C) The corresponding time plot, with predator in green and prey in orange. (D) Coexistence of predator-prey oscillations with a prey-only fixed point. Two trajectories in $p-n$ space for parameters corresponding to the green dot in $a$ with two different sets of initial conditions. In one case ( $n_{t=0}=p_{t=0}=0.5$, dark line) the system displays sustained oscillations whereas in the other case $\left(n_{t=0}=1, p_{t=0}=0.1\right.$, purple line), it undergoes a large excursion before settling on a fixed point with extinction of the predator. (D) The corresponding time plots for preys (yellow) and predators (green) are shown with solid (sustained oscillations) and dashed (evolution toward a stable fixed point) lines.
