Figure S1: Full reaction network of the predator-prey system.
Gray and black parts in the strand sequences emphasize the building of the predator palindrome, not sequence domains. The red star represents the template-bound fluorescent dye and the modlulation of its intensity depending on the hybridization state (single strand versus double strand) of the corresponding extremity of the template. Reaction (11) does not participate in the dynamics of the system, but is included to explain the contribution of the predator to the prey fluorescence.


Figure S2: Experimental limit-cycle for two-colors $\mathrm{PP}_1$ system with different initial conditions. Experiment done in standard conditions (as Fig. 2 of the main text). Up: $G_1=120 n M$; bottom: $G_1=140 n M$; Left: $N_{1, t=0}=10 n M, P_{1, t=0}=30 n M$, right: $N_{1, t=0}=30 n M, P_{1, t=0}=90 n M$. All the trajectories show the experimental evolution of the system from $t=0$ to $t=1600 \mathrm{~min}$. Independently of the initial conditions, the system evolve towards the same stable orbits, which, for given buffer and enzymes conditions, are imposed solely by the template concentration.


1 Methods
In the following discussion, italicized $\mathrm{X}(X)$ denotes the concentration of specie X (e.g., $P_1$ is the concentration of predator $\mathrm{P}_1$ ). "Bst" or "Polymerase" refer to Bst polymerase large fragment and "nicking enzyme" to Nb.BsmI (another nicking enzyme, Nt.BstNBI, is used solely in Section S4-Prey amplification for analytical purposes). PredatorPrey will be abbreviated PP ( $\mathrm{PP}_1$ for the $\mathrm{N}_1, \mathrm{P}_1, \mathrm{G}_1$ system and $\mathrm{PP}_2$ for the $\mathrm{N}_2, \mathrm{P}_2, \mathrm{G}_2$ system).

Buffer and chemical
Bst polymerase and the nicking enzymes were obtained from New England Biolabs (NEB). A working dilution of Bst was prepared by dissolving the original stock solution 5 times in diluent A (NEB) complemented with $0.1 \%$ Triton-X100 and stored at $-20^{\circ} \mathrm{C}$ for further use. Bst was used at $1.87 \%$ of this diluted stock, i.e. $3 \mathrm{unit} / \mathrm{mL}$ or 3.7 nM in the final mix, unless otherwise specified.
$\mathrm{Nb} . \mathrm{BsmI}$ was used at $6 \%$ of the stock solution in the reaction mix (i.e. $600 \mathrm{units} / \mathrm{mL}$ ). This high concentration reflects the facts that nicking enzymes in general are relatively slow catalysts[1]. Here we tried to increase the rate of the nicking reaction as much as possible to simplify the kinetic analysis (see Section S3). In the present case, we did not observe the previously reported inhibiting effect of high concentration of nicking enzymes on polymerase activity[1], presumably because the nicked products are unstable (see Table S2) and melt away very quickly, releasing the enzyme from a potential product inhibited form.

For the degradation reaction we used $t t$ RecJ, the thermophilic equivalent of E. coli's RecJ enzyme from Thermus Thermophilus $[2,3]$, which is a $5^{\prime}->3^{\prime}$ processive exonuclease. The purified enzyme was a kind gift from Ryoji Masui. A working dilution at $2.5 \mu M$ was formed in diluent D (NEB) and stored at $-20^{\circ} \mathrm{C}$. This enzyme was used at 32.5 nM for single PP pair experiments and 15 nM for multispecies systems. Other concentrations, used for kinetic measurements (Fig. 1 and Section S4) are specified when necessary.

Oligonucleotides were obtained from Biomers, with HPLC purification. To avoid any chemical discrepancy between synthetic oligonucleotides and those obtained in situ from enzymatic activity, the prey sequences $\mathrm{N}_1$ and $\mathrm{N}_2$ were ordered with 5' phosphate modification (in the course of the in vitro reaction, new preys are obtained from the nicking of a longer strand, and hence bear a 5 '-phosphate group, whereas standard, unmodified synthetic oligonucleotides are $5^{\prime}-\mathrm{OH}$ ). While this difference is unlikely to have an influence on the PP oscillating reactions, it may be more important in the case of the quantitative measurements of prey concentrations performed by isothermal re-amplification as described in Section S4-Prey amplification.

Throughout the study, we have used a buffer containing 20 mM Tris- $\mathrm{HCl}, 10 \mathrm{mM}\left(\mathrm{NH}_4\right)_2 \mathrm{SO}_4, 10 \mathrm{mM} \mathrm{KCl}, 50 \mathrm{mM} \mathrm{NaCl}, 8 \mathrm{mM} \mathrm{MgSO}{ }_4, 400 \mu \mathrm{M}$ each dNTP (NEB), $0.1 \%$ Synperonic F108 (Aldrich), $5 n g / \mu L$ extremely thermostable single strand binding protein (ETSSB; NEB), $100 \mu \mathrm{~g} / \mathrm{mL}$ BSA (NEB), $2 \mu M$ Netropsin (Sigma Aldrich), 4 mM DTT, $1 \times$ EvaGreenÂ®, $0.5 \times$ ROX (from a standard $100 \times$ dilution provided by Invitrogen) and a pH of 8.8 . A $4 \times$ stock solution of this buffer was prepared (excluding BSA, DTT and ETSSB, which were added, together with enzymes and oligonucleotides, during reaction assembly) and used afterward.

In the following, "standard conditions" therefore refers to this buffer, $\mathrm{T}=46.5^{\circ} \mathrm{C}$, Bst $3.7 \mathrm{nM}, \mathrm{Nb} . \mathrm{BsmI} 600$ units/mL, ttRecJ 32.5 nM .

The functions of the components included in this buffer are summarized below.
- BSA, DTT and Synperonic F108 (a surfactant) act as stabilizing agents for the enzymes. Their inclusion in the buffer allows us to run experiments for up to a few days in closed tubes without excessive loss of enzymatic activity (whereas for example Nb.BsmI is reported by NEB to be stable for less than 8 hours in the recommended digestion buffer, at the recommended temperature).
- EvaGreen is a DNA intercalating green fluorescent dye that does not inhibit enzymatic activity. It has a marked preference for duplex DNA over single strands oligonucleotides. In a PP oscillator, it allows us to observe nonspecifically the appearance/disappearance of double strand DNA. However we have also used the lower signal generated by single strand binding for some kinetic measurement (digestion rate of $\mathrm{N}_1$ and $\mathrm{N}_2$ ). As other intercalating agents, EvaGreen has an effect on DNA melting temperature. Hence the melting temperatures used in Section S4-Predation were measured in its presence.
- Netropsin is a short peptide with a strong affinity for AT-rich sequences in double-stranded DNA. We use it to prevent the occurrence of parasitic DNA species, which tend to appear stochastically when mixtures of polymerase and restriction enzymes are incubated for extended periods of time[4]. Such species emerging and exponentially developing- in isothermal amplification schemes have the potential to disrupt experimental setups[5] or even to contaminate a whole lab[6] and maximum precautions[7] should be taken to minimize their emergence and spreading. The analysis of such parasites induced by polymerase-nicking enzyme mixes by Tan et al. [4] has shown that they typically contain long AT repeats, interspaced by sequences that serves as target of the restriction enzyme present in the buffer. We observed that netropsin tends to delay the development of such species in a contaminated experiment, while having at most a marginal effect on the targeted oscillatory reaction. As a rationale, it may be tentatively proposed that by binding on AT rich sequences, netropsin locally stabilizes the double helix[8] and inhibits the strand slippage and hairpin-coil transition mechanisms that is necessary for the growth of the parasitic specie[4]. However, we have also noticed that a variety of parasitic reactions with exponential fluorescence signatures can appear after long incubation of PP mixtures (even in the presence of netropsin), some of them even being able to coexist with the oscillatory PP reaction.
- ETSSB is a thermostable single-stranded DNA binding protein with a large binding site (from 35 to 65 bases, depending on salt conditions, in the case of E. Coli SSB[9]). SSB has been reported as having a facilitating effect in nicking enzyme mediated DNA amplification[1]. In our case, we have noticed that ETSSB accelerates the growth rate of the prey, possibly by lowering the melting temperature of the G template hairpin. We also observed that ETSSB has an accelerating role in the growth of $\mathrm{P}_1$ predators, less marked for $\mathrm{P}_2$.

Reaction assembly
Reactions were typically run in parallel in strips of $20 \mu \mathrm{~L}$ PCR white tubes. A master mix was first assembled by mixing all the common components at $4^{\circ} \mathrm{C}$. Concentrations of chemicals that were different from tube to tube (e.g. the concentration of template in the experiment shown in Fig. 3d of the main text, or the initial concentrations of prey and predators in the experiment of Fig. 4e), were assembled in a minimal reasonable volume directly in the PCR tube, after which the master mix containing all other components was added. This procedure insures minimum dispersion in the concentration of the fixed components from tube to tube ${ }^1$. Careful mixing was applied ( 10 s vortex) at each assembly stage. The tubes were then quickly centrifuged and transferred in a Bio-Rad CFX or MiniOpticon thermocycler set at a constant temperature of $46.5^{\circ} \mathrm{C}$ (this temperature was selected somewhat arbitrarily, as the oscillating PP reaction can be observed over a range of at least several degrees, provided the enzymes and template concentrations are adjusted).

Monitoring and analysis
The fluorescence records were obtained from the real-time thermocyclers at interval varying from a few seconds (for kinetic measurements described in Section 4) to several minutes (for the longest experiments, e.g. Fig. 3b or Fig. 4). The fluorescence cross-talks between the different channels (green, yellow and deep red) were removed by the built-in software after calibration for EvaGreen (ex. 500 nm , em. 530 nm ), Dy530 (ex. 539 nm , em. 561 $\mathrm{nm})$, JOE (ex. 520 nm , em. 548 nm ) and Dy681 (ex. 691 nm , em. 708 nm ). A third order polynomial base line was subtracted from these crude data sets to eliminate the background fluorescence and long term drifts whose causes were obviously not relevant to the reactions of interest (physical effects, evaporation, dye bleaching, slow side reactions, etc. ..). In the case of Dy681 the fluorescence ratio Dy681/ROX was used instead because it tended to be more stable.

The green fluorescence produced by EvaGreen contains a contribution from each of the species shown in Fig. S1. Because of the double-strand preference of EvaGreen, the signal is dominated by the contribution of double-stranded species. Free preys are shorter than predators and don't form stable hairpins or duplex, hence they produce less fluorescence. Duplexes including the template (template-prey but also possibly non-productive template-predator) may also contribute to the signal, but will be at least partially quenched in the green channel by the presence of the template-bound dye of higher wavelength. Rather than trying to obtain the fluorescent contribution of each molecular species (which can be dependent on the experimental conditions and optical set-up), we instead define the green signal as a linear combination of the prey and predator concentrations, dominated by the latter. Following a similar argument (but considering only the species involving the template), we define the yellow (or red) signal as a linear combination dominated by the prey concentration and containing a predator contribution (see reaction (11) in Fig. S1). Because of the linearity, we directly obtain that predator and prey are also linear combinations of the green and yellow signal. Hence, in arbitrary fluorescent units, we just need to find out the cross-talk of predators in the mostly preys signal (yellow or red) and that of the preys in the mostly predators signal (green). These parameters are determined empirically by minimizing long tails for preys spikes and pre-bumps for predators spikes. This process is applied at once to one set of experiment, for example all the plots of Fig. 3d,e share the same cross-talk parameters (in this specific case, the cross-talk of predators to preys was found at $28 \%$ and that of prey to predators was found at $22 \%$ ). Cross-talks talks for the $P P_2$ pair were typically lower. These corrected signals correspond to the "Green (Predator) shift" and the "Yellow (Prey) shift" (or "Red (Prey) shift") plotted for experiments with a single PP pair and the symbiotic experiments of Fig. 5a. In the case of multiple PP pairs, this procedure cannot be applied since we observe only three signals for four species. Hence in this case we only remove the preys contribution to the $P_1+P_2$ signal (i.e. we don't correct for the small contributions of the predators to their respective prey signals).


${ }^1$ When comparing two experiments that were not created from the same master mix (e.g.- experiments performed on two different days) one should assume some dispersion on the concentrations, as a result of unavoidable uncertainties on pipetted volumes. For reaction performed at an interval of several weeks, a change of enzymatic activity, due to the aging of the enzyme stocks should be further expected.



2 Sequence design
The design of the basic PP reaction network comes down to solely selecting the sequence of the G template, in such a way that the reactivity of the system organizes itself along the reaction network depicted in Fig. S1.

This mapping can be reduced to a short set of design constraints:
D Predators should have a palindromic sequence because of their templated duplication mechanism (reaction 7).
- Moreover, this palindrome should be as short as possible so that predator duplexes significantly dissociate (reaction 8 ) -avoiding product inhibition. While this requirement may seem contradictory with the fact that the prey needs to form a primer-template with the predator for the reaction to proceed (reaction 6 ), we have observed that Bst polymerase is able to extend efficiently such short and unstable partial duplexes, even when working well above their melting temperature.
- In order to limit the length of the predator, a part of the palindromic sequence was already included in the prey's 3' end sequence. If this palindrome is too long however, preys may become able to form so-called primerdimers because they possess an overlapping self-complementary sequence. This may lead to an unwanted leak reaction if such primer-dimers are extended by the polymerase. After some trial-and-errors, we found that a length of 10 bases for the prey, 6 of which are palindromic, and hence a 14 base predator was appropriate for predator growth in the $40-50^{\circ} \mathrm{C}$ temperature range, without leading to a significant amount of the afore mentioned leak reaction (at least for prey concentrations in the hundred of nano-molar range, see also Section S4-Prey amplification).
- The prey should further incorporate the recognition sequence of a nicking enzyme, itself stable and active over the working temperature range (i.e. thermophilic). This is necessary for the prey's extension-nicking replication mechanism (reaction 3 and 4).

D Prey amplification and predator growth reactions are compatible (in the same mix) at the cost of one further constraint: the nicking enzyme recognition site should be such that a single prey sequence does not contain the recognition site (only a dual repeat of prey does). Otherwise this sequence would be transferred to the predator and would render them sensitive to being nicked -back to prey. To satisfy this constraint we selected $\mathrm{Nb} . \mathrm{BsmI}$ as the nicking enzyme for this work. ${ }^2$. It recognizes the double stranded sequence $5^{\prime}-\mathrm{G}^{\wedge}$ CATTC-3' and cuts within this site, at the position marked by a hat $\left({ }^{\wedge}\right)$. Hence this recognition sequence, in double strand form, is found only on G duplexes and nowhere else in the system (reaction 4).
- Finally the template sustaining prey growth should be stable against the exonuclease, but not preys themselves, nor predators (reactions 9 and 10). As already shown[10], this can be achieved with a few phosphorothioates backbone modifications.

The sequences are then decided by combining these constraints. The positions that remained free (unconstrained) in the sequence are assigned random bases and we selected two individual (Table S1) with a prey-template melting temperature (at 100 nM ) slightly below the working temperature (Table S2).

Additional, non-genetic modifications of the template were used to protect it from exonuclease degradation, block its extension by the polymerase and use it as a fluorescent reporter of prey concentration:
- Since ttRecJ is a processive exonuclease digesting from the 5' end of single-strand oligonucleotides, we expected that phosphorothioates backbone modifications on this side of the template should provide a good protection[11]. Indeed templates bearing three modifications in a row were almost insensitive to ttRecJ degradation. We also verified that this modification did not significantly affect polymerase or nicking enzyme activity on the template.
- In the targeted reaction set the $3^{\prime}$ end of the template is never in a position to serve as a primer for the polymerase (and hence to be extended). Still we took the precaution to block this end with a phosphate modification. Since only $3^{\prime} \mathrm{OH}$ ends can be extended by DNA polymerases, this precaution serves to prevent potential leak reactions.
- For templates used as prey reporter, this terminal 3 '-phosphate was replaced by a fluorescent dye. The mechanism by which such modified templates report on their input strand (prey) concentration is detailed elsewhere[12]. It should be noted that these modifications of the template 3' end may not be neutral with respect to the kinetic parameters of the system (it will slightly change the associated melting temperatures[13], but may also affect the recognition by some enzymes). We observed indeed that template 3 ' modifications had an impact on the dynamic of the system. In practice however, we found that they could be compensated for by varying the concentration of template or the reaction conditions.

${ }^2$ The use of this enzyme is recommended at $65^{\circ} \mathrm{C}$ but we found that it retain a correct activity -at least fo the nicking of short
duplexes like the one of interest here- down to $40{ }^{\circ} \mathrm{C}$


The sequence of Sy was chosen at random (after checking the absence of cross-talks with the previously designed species), and contain the nicking enzyme recognition sequence. The sequences of the symbiotic templates $\mathrm{G}_{2 \rightarrow \mathrm{Sy}}$ and $\mathrm{G}_{\mathrm{Sy} \rightarrow 2}$ (as well as $\mathrm{G}_{2 \rightarrow 1}$ and $\mathrm{G}_{1 \rightarrow 2}$ ) were then simply formed by combination of the corresponding input and output sequences.

Mes $\mathrm{N}_1$ and $\mathrm{MesN}_2$ are used in Section S4-Prey amplification to quantitatively measure the concentrations of $\mathrm{N}_1$ and $\mathrm{N}_2$, respectively. These re-amplification templates incorporate the recognition sequence of the nicking enzyme Nt.BstNBI[4].


\begin{table}
\captionsetup{labelformat=empty}
\caption{Table S1: Sequences and modification of the oligonucleotides used in this study.}
\begin{tabular}{|l|l|l|}
\hline Name & Sequence ( $5^{\prime} \rightarrow 3$ ') & 3'-modifications \\
\hline $\mathrm{N}_1$ & CATTCGGCCG & \\
\hline $\mathrm{P}_1$ & CATTCGG-CCGAATG & \\
\hline $\mathrm{G}_1$ & C*G*G*CCGAATG-CGGCCGAATG & phos or Dy530 (yellow fluorescence) \\
\hline $\mathrm{N}_2$ & CATTCCCGGG & \\
\hline $\mathrm{P}_2$ & CATTCCC-GGGAATG & \\
\hline $\mathrm{G}_2$ & C*C*C*GGGAATG-CCCGGGAATG & phos or Dy681 (deep red fluorescence) \\
\hline Sy & CATTCGGGCG & \\
\hline $\mathrm{G}_{1 \rightarrow 2}$ & C*C*C*GGGAATG-CGGCCGAATG & phos \\
\hline $\mathrm{G}_{2 \rightarrow 1}$ & C*G*G*CCGAATG-CCCGGGAATG & phos \\
\hline $\mathrm{G}_{2 \rightarrow \mathrm{Sy}}$ & C*G*C*CGAATG-CCCGGGAATG & phos \\
\hline $\mathrm{G}_{\mathrm{Sy} \rightarrow 2}$ & C*C*C*GGGAATG-CGCCCGAATG & JOE (yellow fluorescence) \\
\hline $\mathrm{MesN}_1$ & CGGCCGAATG-CACTGACTCGGCCGAATG & \\
\hline $\mathrm{MesN}_2$ & CCCGGGAATG-CACTGACTCCCGGGAATG & \\
\hline
\end{tabular}
\end{table}
* indicates a phosphorothioate backbone modification used to block exonuclease digestion.
The area in bold is the recognition sequence of the nicking enzyme Nb.BsmI (or Nt.BstNBI in the case of $\mathrm{MesN}_1$ and $\mathrm{MesN}_2$; see Section S4-Prey amplification kinetic).

\begin{table}
\captionsetup{labelformat=empty}
\caption{Table S2: Melting temperatures for hybridization of the oligonucleotides in $\mathrm{PP}_1$ and $\mathrm{PP}_2$.}
\begin{tabular}{|l|l|l|l|l|l|l|l|}
\hline & $\mathrm{N}_1$ & $\mathrm{P}_1$ & $\mathrm{G}_1$ & $\mathrm{N}_2$ & $\mathrm{P}_2$ & $\mathrm{G}_2$ & Folding \\
\hline $\mathrm{N}_1$ & $28.3^{\circ} \mathrm{C}$ & $43.4^{\circ} \mathrm{C}$ & $43.8^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $0.1^{\circ} \mathrm{C}$ & $0.9^{\circ} \mathrm{C}$ & $<0{ }^{\circ} \mathrm{C}$ \\
\hline $\mathrm{P}_1$ & & $54.0^{\circ} \mathrm{C}$ & $46.9{ }^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $5.0^{\circ} \mathrm{C}$ & $5.9^{\circ} \mathrm{C}$ & $51.1^{\circ} \mathrm{C}$ \\
\hline $\mathrm{G}_1$ & & & $64.4^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $<0{ }^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ & $91.6^{\circ} \mathrm{C}$ \\
\hline $\mathrm{N}_2$ & & & & $23.4{ }^{\circ} \mathrm{C}$ & $40.7^{\circ} \mathrm{C}$ & $41.2{ }^{\circ} \mathrm{C}$ & $<0^{\circ} \mathrm{C}$ \\
\hline $\mathrm{P}_2$ & & & & & $52.4^{\circ} \mathrm{C}$ & $44.5^{\circ} \mathrm{C}$ & $51.1^{\circ} \mathrm{C}$ \\
\hline $\mathrm{G}_2$ & & & & & & $61.5^{\circ} \mathrm{C}$ & $91.0^{\circ} \mathrm{C}$ \\
\hline Sy & & & $36.5^{\circ} \mathrm{C}$ & & & $15.3^{\circ} \mathrm{C}$ & \\
\hline
\end{tabular}
\end{table}
The area in white shows the intra-pair interactions while the area in gray reports the inter-pair interactions. The blue column describes the folding reactions. Calculated with Dynamelt (http://mfold.rna.albany.edu/?q=DINAMelt) for 100 nM concentrations of each oligos and salt conditions as in experimental setup. Note that the presence of ETSSB and the intercalating dye EvaGreen will affect the actual melting temperatures.



3 A simple two species model obtained from the chemical mechanism
In this section we build a simple, two-variable mathematical model. In the process, we try to maintain as much as possible a chemically reasonable mechanistic description with respect to the chemical reaction network shown in Fig. S1. The goal is in particular to preserve a link between the parameters of this high-level two variables model, and measurable kinetic and thermodynamic parameters of each of the individual, lower-level chemical interactions that compose this predator-prey system.

Simplifications used to reach this model are based on the fact that the polymerase and the exonuclease are processive enzymes, so that even if they perform complicated multistep chemical transformations (e.g. in the case of the polymerase-mediated reaction (3) of Fig. S1, the sequential selection and polymerization of the 10 matching bases necessary to extend a G:N partial duplex to its full length), no intermediate product accumulates and these reactions can be considered as individual chemical events. Another simplifying principle is that, because G and P form hairpins or palindromic duplexes, and considering the relatively high working temperature, their binding to the prey is weak. It thus becomes possible to consider that the amount of duplex species ( $\mathrm{P}: \mathrm{N}$ or $\mathrm{G}: \mathrm{N}$ ) is always a small fraction of the total population of each P and N species.

We denote the concentration of prey as $N$, that of the template as $G$, the concentration of prey-template partial duplex as $G: N$, the concentration of predators as $P$, that of prey-predators partial duplexes $P: N$, the concentration of Bst polymerase as pol and the concentration of ttRecJ enzyme as rec. Other notations are introduced when necessary.

Prey Growth
The prey amplification is a composite process comprising reaction (1) to (5) in Fig. S1.
With an estimated $K_a^{G N}$ of $5 \times 10^{-7} \mathrm{nM}^{-1}$, the association of $\mathrm{N}_1$ and $\mathrm{G}_1$ is very unstable. For example, at 100 nM of each $\mathrm{N}_1$ and $\mathrm{G}_1$, NUPACK (http://www.nupack.org/) predicts less than 15 pM of $\mathrm{N}_1$ : $\mathrm{G}_1$ duplex at the working temperature. We can then assume that the extension of these few partial duplexes by the polymerase will be a first order and rate-limiting step of the amplification process. The same holds for the second predator-prey system ( $\mathrm{PP}_2$ ).

However this extension is followed by a nicking event. This enzymatic mechanism will impose an upper limit on the maximum rate of the process, that we choose to describe empirically with a Michaelis-Menten-like expression. We then write the corresponding flux (i.e. the kinetic expression corresponding to this specific process) as:
$$
\varphi_{N \triangleright N}=k \cdot \operatorname{pol} \frac{G: N}{K+G: N}
$$

Note that the values of $k$ and $K$ should be expected to depend on the nicking enzyme, but because we have kept it fixed throughout this study, we consider these parameters as constants (see Section S6 for discussion on this point).

From reaction (1) in Fig. S1, and considering a fast equilibrium, we have:
$$
G: N=K_a^{G N} \cdot G_{\text {free }} \cdot N_{\text {free }}
$$

The prey species N is involved in a number of equilibria, but all of them have a small binding constant so we can assume that $N_{\text {free }} \approx N_{\text {total }} \approx N$. Similarly, and assuming that the template doesn't accumulate too much in the doubled stranded form, $G_{\text {free }} \approx G_{\text {total }} \approx G$. We can then write:
$$
\varphi_{N \triangleright N}=k \cdot \operatorname{pol} \frac{K_a^{G N} \cdot G \cdot N}{K+K_a^{G N} \cdot G \cdot N}=k_1 \cdot \operatorname{pol} \frac{G \cdot N}{1+b \cdot G \cdot N}
$$
with $k_1=K_a^{G N} \cdot k / K$ and $b=K_a^{G N} / K$.
Predation
This process encompasses reactions (6) to (8) in Fig. S1. We note again that the equilibrium of association of the prey with the predator, (reaction 6), has a low binding constant and is expected to be mostly displaced toward dissociation. At $46.5^{\circ} \mathrm{C}$ and 100 nM of each $\mathrm{P}_1$ and $\mathrm{N}_1$, NUPACK predicts less than 8 nM of P:N duplex. Then, at any time in the reaction, only a small amount of prey-predator duplex forms. If further reaction (8) is fast enough ${ }^3$, the extension to predator duplex should be a first-order and rate-limiting step for the overall process. With these assumptions and the above notations, the duplication of predator, with consumption of one prey give the rise to the flux:
$$
\varphi_{N \triangleright P}=k^{\prime} \cdot \operatorname{pol} \cdot P: N
$$

From reaction (6) in Fig. S1, we have:
$$
P: N=K_a^{P N} \cdot P_{\text {free }} \cdot N_{\text {free }}
$$

As before we have: $N_{\text {free }} \approx N_{\text {total }} \approx N$. We assume a similar relation for the predator. Then, defining $k_2=k^{\prime} \cdot K_a^{P N}$, we obtain:
$$
\varphi_{N \triangleright P}=k^{\prime} \cdot K_a^{P N} \cdot \mathrm{pol} \cdot P \cdot N=k_2 \cdot \mathrm{pol} \cdot P \cdot N
$$

As can be seen in Fig. S9 this simple rate equation is able to describe accurately the predator replication process, at least for concentrations in the hundred of nano-molar range. This supports at posteriori the relevance of the previous approximations. For a more complete treatment of palindromic replicators dynamics, see [14].

Degradation
The exonuclease that we use is a processive enzyme and hence the degradations of both prey and predator correspond to a single-step enzymatic reaction which can be reasonably described by a Michaelis-Menten kinetic expression (eq. 2).
$$
\frac{d X}{d t}=r e c \cdot k_{c a t, X} \frac{X}{K_{m, X}+X}=V_m \frac{X}{K_{m, N}+X}
$$

We note $k_{\text {cat }, N}$ and $k_{\text {cat }, P}$ ttRecJ catalytic rates versus N and P respectively, and $K_{m, N}$ and $K_{m, P}$ the corresponding Michaelis-Menten constants. Because prey and predator use the same enzyme for their degradation, there is a competitive situation, which leads to the two coupled flux equations:
$$
\begin{aligned}
& \varphi_{N \triangleright \varnothing}=\operatorname{rec} \cdot k_{c a t, N} \frac{N}{K_{m, N}\left(1+\frac{N}{K_{m, N}}+\frac{P}{K_{m, P}}\right)} \\
& \varphi_{P \triangleright \varnothing}=\operatorname{rec} \cdot k_{c a t, P} \frac{P}{K_{m, P}\left(1+\frac{N}{K_{m, N}}+\frac{P}{K_{m, P}}\right)}
\end{aligned}
$$

However, considering that $K_{m, N} \gg K_{m, P}$ (see Table S4 in Section S4) and that $\max [N(t)] \approx \max [P(t)] \approx K_{m, P}$ (e.g. Fig. 2 g main text) we can approximate these expressions as:
$$
\begin{aligned}
\varphi_{N \triangleright \varnothing} & =r e c \cdot k_{c a t, N} \frac{N}{K_{m, N}\left(1+\frac{P}{K_{m, P}}\right)} \\
\varphi_{P \triangleright \varnothing} & =r e c \cdot k_{c a t, P} \frac{P}{K_{m, P}\left(1+\frac{P}{K_{m, P}}\right)}
\end{aligned}
$$

Defining the first-order degradation rate for N and P as $k_N=\frac{k_{\text {cat }, N}}{K_{m, N}}$ and $k_P=\frac{k_{\text {cat }, P}}{K_{m, P}}$ we get:
$$
\begin{gathered}
\varphi_{N \triangleright \varnothing}=r e c \cdot k_N \frac{N}{1+\frac{P}{K_{m, P}}} \\
\varphi_{P \triangleright \varnothing}=r e c \cdot k \frac{P}{1+\frac{P}{K_{m, P}}}
\end{gathered}
$$
${ }^3$ ETSSB may also play a role at this stage


Complete system
The kinetic evolution of the prey species is determined by the contribution of growth, predation and degradation reactions, while that of the predator results from predation and degradation. Hence the complete system is dynamically described by the set of two coupled ordinary differential equations (3) and (4).
$$
\begin{gathered}
\dot{N}=\frac{d N}{d t}=k_1 \cdot \mathrm{pol} \cdot G \frac{N}{1+b \cdot G \cdot N}-k_2 \cdot \mathrm{pol} \cdot N \cdot P-\mathrm{rec} \cdot k_N \frac{N}{1+\frac{P}{K_{m, P}}} \\
\dot{P}=\frac{d P}{d t}=k_2 \cdot \mathrm{pol} \cdot N \cdot P-\mathrm{rec} \cdot k_P \frac{P}{1+\frac{P}{K_{m, P}}}
\end{gathered}
$$

We define: $\tau=t / t_c=t \cdot k_2 \cdot \mathrm{pol} \cdot K_{m, P}, n=N / K_{m, P}, p=P / K_{m, P}, g=G / G_0=\frac{k_1 \cdot G}{k_2 \cdot K_{m, P}}, \lambda=k_N / k_P$, $\beta=b \cdot k_2 \cdot K_{m, P}^2 / k_1$, and $\delta=\frac{r e c}{p o l} \cdot \frac{k_P}{k_2 \cdot K_{m, P}}$ and obtain the nondimensionalized equations:
$$
\begin{gathered}
\frac{d n}{d \tau}=\frac{g \cdot n}{1+\beta \cdot g \cdot n}-p \cdot n-\lambda \cdot \delta \frac{n}{1+p} \\
\frac{d p}{d \tau}=p \cdot n-\delta \frac{p}{1+p}
\end{gathered}
$$

Hence, our chemical construction can be modeled with a rather simple version of the predator-prey system of equations. Despite its extreme simplicity, we will see that this model is able to reproduce accurately the experimental observations, for two species systems (Fig. 2 and Figs S13, S14, S15) as well as for more complex multi-species systems (Fig. 3 and 4, Section S6). The relatedness between our model -derived from a simple kinetic description of the chemical reactions- and classic models of theoretical ecology permits a meaningful use of the ecological metaphor to understand the functioning of an abiotic system -and vice versa. Also note that a large body of theoretical work on the molecular origin of life[15] has been built on the "replicator equation"[16] itself analogous[17] to Lotka-Volterra systems.


Bifurcation analysis
Using the optimized values of the kinetic parameters obtained in section S4 and summarized in Table S5 for standard conditions ( $r e c=32.5 n M$ ), we can calculate the non-dimensional parameters and the scaling factors $G_0$ and $t_c$ for standard conditions. They are gathered in Table S3.

\begin{table}
\captionsetup{labelformat=empty}
\caption{Table S3: Nondimensional parameters of the model for standard conditions}
\begin{tabular}{|c|c|c|}
\hline & $\mathrm{PP}_1$ & $\mathrm{PP}_2$ \\
\hline \hline$\beta$ & $8.7 \times 10^{-2}$ & $1.3 \times 10^{-1}$ \\
\hline$\lambda$ & 4.5 & 3.7 \\
\hline$\delta$ & 0.39 & 0.38 \\
\hline$G_0(\mathrm{nM})$ & 53 & 86 \\
\hline$t_c(\mathrm{~min})$ & 2.6 & 3.6 \\
\hline
\end{tabular}
\end{table}

We will now study the behavior of this system as a function of $g$, the scaled template concentration, which can be regarded as the richness of the environment sustaining prey growth, the enzymatic ratio $\delta$, which gives the scaled degradation rate of the predator, and $\lambda$, the relative first-order degradation rate of the prey versus that of the predator. In the following discussion, we study the necessary conditions for the emergence of oscillations in this model.


First-order exonuclease (large $K_{m, P}$ )
In a first step, we study the case where $P$ is very small compared to $K_{m, P}$ at any time, i.e. we put ourselves in the case where the predator never saturates the exonuclease. With this simplification, the stability of the system of eqs. 7 and 8 can be analytically obtained.

Because $p \ll 1$ is always true, the equations become:
$$
\begin{gathered}
\dot{n}=\frac{g \cdot n}{1+\beta \cdot g \cdot n}-p \cdot n-\lambda \cdot \delta \cdot n \\
\dot{p}=p \cdot n-\delta \cdot p
\end{gathered}
$$

We first note that the case of negligible values of $\beta$ (i.e. unrealistic never-saturating production), would bring us to the pure Lotka system, with its pathologic conservative orbits. This classic but physically irrelevant problem has been well studied (see for example[18, 19])

For a realistic system, i.e. with limited grow potential, hence non-negligible values of $\beta$, we show in the following that the system always has a stable fixed point and can not display limit cycles.

By setting $\dot{n}$ and $\dot{p}$ to 0 , one finds three equilibrium points:
$\triangleright(0,0)$ i.e extinction of both species,
$\triangleright\left(\frac{g-\delta \cdot \lambda}{\beta \cdot \delta \cdot \lambda \cdot g}, 0\right)$ i.e. extinction of the predator and stable population of prey, valid for $\lambda \cdot \delta<g$,
$\triangleright\left(\delta, \frac{g}{1+\beta \cdot \delta \cdot g}-\lambda \cdot \delta\right)$, corresponding to coexistence of both species, valid for $g>\lambda \cdot \delta /\left(\beta \cdot \lambda \cdot \delta^2-1\right)$.
Using linear stability analysis (i.e. by looking at the eigenvalues of the community matrix linearized around these points[18]) we find that each of these fixed points is stable for, respectively, $\lambda \cdot \delta>g, \lambda \cdot \delta<g<\lambda \cdot \delta /\left(\beta \cdot \lambda \cdot \delta^2-1\right)$ and $g>\lambda \cdot \delta /\left(\beta \cdot \lambda \cdot \delta^2-1\right)$. Therefore the system can have three behaviors, depending on the relative values of the parameters: for small values of $g$ it evolves toward the $(0,0)$ fixed point (absence of any species). For intermediate values of $g$, it evolves toward a stable state where the steady prey population is not high enough to support a predator population. Finally, for large $g$, predator and prey reach a stable coexistence steady state, possibly after some damped oscillations. It can be seen that this simple model always has a single stable equilibrium for any set of parameters and that no sustained oscillations are permitted (bistability is excluded as well). This system can at most display damped oscillations (see Fig. S3). This behavior (i.e. the nonexistence of sustained oscillations) is a consequence of the saturation of prey growth. Such a stabilizing effect of limited growth potential has already been noticed[19] and the absence of cycle in a similar model with limited prey growth (the Volterra model) been noted[20, 21].

In classical PP models that do display a limit cycle, this stabilizing effect is generally balanced by nonlinearities in the functional response (i.e. the predation term). In our model the functional response is linear. We see below (section S3-General case) that in the present system, it is the saturable decay term that allows stable limit cycles to occur.

In relation with ecological modeling, one may note at this stage that, in the absence of predators, eq. (7) can be rewritten as:
$$
\frac{g \cdot n}{1+\beta \cdot g \cdot n}-\lambda \cdot \delta \frac{n}{1+p}=\frac{(g-\lambda \cdot \delta)}{(1+\beta \cdot g \cdot n)} n\left(1-\frac{\beta \cdot \lambda \cdot \delta \cdot g}{g-\lambda \cdot \delta} n\right)=G R(n) \cdot n \cdot(1-C P \cdot n)
$$

Thus the prey growth in our system has a form close to the logistic equation, one of the most used equation in ecology to describe saturable autocatalytic growth processes. By analogy, the carrying capacity would be $C P= \frac{g-\lambda \cdot \delta}{\beta \cdot \lambda \cdot \delta \cdot g}$, and the growth rate $G R(n)$. $G R(n)$ is given at low prey concentrations $(n \ll 1 /(\beta \cdot g))$ by $g-\lambda \cdot \delta$ and tends to decrease at higher concentration, giving a slightly less pronounced sigmoidal shape to the population growth (see Fig. S4).

Figure S3: Dynamics of the system with first order degradation.
Simulations using eq. (5) and (6). Unspecified parameter values are from Table S3-PP ${ }_1$. (A) Bifurcation diagram in the $g$ - $\lambda$ parameter space, showing the region of existence for the 3 stable equilibria (i.e. species eventually reaching extinction or a steady state noted $X_{s s}$ ). (B) Typical trajectory (dashed line) showing damped oscillations in the $p-n$ space for parameters corresponding to the red spot in $a$ (i.e. in the region of stable coexistence of prey and predator). The prey and predator nullclines are shown in red and blue, respectively. (C) The corresponding time plot, in nondimensional time and concentrations (prey in yellow, predator in green).

Figure S4: Prey growth in a predator-free world. Present chemical model versus logistic equation.
$\tau$ is the nondimensional time and $n$ the nondimensional prey concentration. $n_{t=0}=10^{-3}$. Solid line: present model (eq. 9); dashed line: logistic equation. The parameters used are those given in Table S 3 for $\mathrm{PP}_1$, and $g=2.5$.

Figure S5: Dynamics of the system with saturated degradation.
Model with competitive Michaelis-Menten degradation (eqs. 5 and 6). Unspecified parameters values are from Table S3-PP ${ }_1$.
(A) Bifurcation diagram in the $g-\lambda$ parameters space, showing the region of existence for stable equilibria and oscillations.
(B) Typical trajectory (dashed line) showing sustained oscillations in the $\mathrm{p}-\mathrm{n}$ space for parameters corresponding to the red spot in $a$ (i.e. in the region of stable oscillations). The prey and predator nullclines are shown in red and blue, respectively. (C) The corresponding time plot, with predator in green and prey in orange. (D) Coexistence of predator-prey oscillations with a prey-only fixed point. Two trajectories in $p-n$ space for parameters corresponding to the green dot in $a$ with two different sets of initial conditions. In one case ( $n_{t=0}=p_{t=0}=0.5$, dark line) the system displays sustained oscillations whereas in the other case $\left(n_{t=0}=1, p_{t=0}=0.1\right.$, purple line), it undergoes a large excursion before settling on a fixed point with extinction of the predator. (D) The corresponding time plots for preys (yellow) and predators (green) are shown with solid (sustained oscillations) and dashed (evolution toward a stable fixed point) lines.


General case (Michaelis-Menten exonuclease)

We now turn to the case of finite value for $K_{m, P}$, i.e. the complete set of equations (5) and (6). The fixed points involving no predator (i.e. those at $(0,0)$ and at $\left(\frac{g-\delta . \lambda}{\beta . \delta . \lambda . g}, 0\right)$ ), as well as their stability are not affected by this modification. However, a new fixed point appears in the coexistence area. The bifurcation diagram showing the effect of saturated degradation is shown in the $g-\lambda$ plane in Fig. S5a. The regions of existence and stability of the coexistence fixed point (including the Hopf bifurcation curve) were obtained analytically after noting that 3.g.n is small compared to 1 for the solutions of interest and thus that eq. (5) can be expanded as:
$$
\dot{n}=g \cdot n(1-\beta \cdot g \cdot n)-p \cdot n-\lambda \cdot \delta \frac{n}{1+p}
$$

For intermediate value of $g$ and small values of $\lambda$, stable oscillations are now predicted (Fig. S5). The promotion of oscillations by a nonlinear, MM degradation term (an enzymatic sink) in various theoretical models of chemical reaction networks has already been reported[22, 23].

In a small part of the diagram, these oscillations may coexist with another, non-oscillating solution (extinction of the predator, or extinction of both the prey and the predator). In this area, the behavior of the system will depend on the initial conditions (Fig. S5d-e). However we have not searched experimentally for such a behavior.

Effect of prey-predator coupling at the degradation stage
One may ask whether the fact that the prey degradation is coupled to that of the predator via competitive inhibition has a strong impact on the dynamic. Fig. S6 shows the bifurcation diagram in the hypothetical case where this coupling is removed (hence pure first order degradation is assumed for the prey and simple Michaelis-Menten for the predator, equations (10) and (11)).
$$
\begin{gathered}
\dot{n}=\frac{g \cdot n}{1+\beta \cdot g \cdot n}-p \cdot n-\lambda \cdot \delta \cdot n \\
\dot{p}=p \cdot n-\delta \frac{p}{1+p}
\end{gathered}
$$

The bifurcation diagram still displays an area for stable oscillations, although somehow shrunk compared to the realistic (competitive) situation. It thus appears that in this model, saturation in the degradation pathway is required for oscillations, but not the competitive degradation between the two species.

Figure S6: Effect of removing the competitive coupling between predator and prey in a single pair.
Here we assume saturated degradation for the predator, but first order degradation for the prey. This correspond to the hypothetical case where prey and predator would not compete for ttRecJ (eqs (10) and (11)). The resulting bifurcation diagram retain the shape of the general (competitive) case, but the area displaying oscillations recedes to smaller values of $\lambda$ (gray area). Bimodal behavior is not predicted anymore.

4 Estimating the kinetic parameters of the model
The two variable mathematical model derived in Section S3 needs to be fed with quantitative values of the parameters. We show in this section that it is possible to obtain a reasonable estimate for these kinetic parameters by studying independently the three interactions -prey amplification, predation and degradation- that compose the PP system. The resulting numerical values are summarized in Table S5.

Degradation: ttRecJ kinetic constants
The kinetic constants for this enzyme relative to its various substrates (both prey and both predators) were evaluated as reported[10] using a continuous assay. This was done in the same buffer as complete PP experiments (in the presence of all components including Nb.BsmI, ETSSB and EvaGreen, but not Bst polymerase), and at the same temperature ( $46.5^{\circ} \mathrm{C}$ ), taking advantage of the fluorescence produced by these DNA species in the presence of EvaGreen. Detailed procedures are given below and results are summarized in Table S4.

Prey degradation
For $\mathrm{N}_1$ and $\mathrm{N}_2, 4 \mu M$ of substrate were incubated in the presence of 8.3 nM of ttRecJ. The fluorescence was recorded at regular intervals. This experimental data sets were plotted as rates versus concentrations and fitted to the Michaelis-Menten equation (eq. 2)

The parameters $k_{\text {cat }}$ and $K_m$ were extracted from the fits for $\mathrm{N}_1$ and $\mathrm{N}_2$ (Table S4). Both accurately described the experimental data (Fig. S7).

Predator degradation
In the case of the predator species P , the DNA strands exist as a mixture of dimers ( $\mathrm{P}: \mathrm{P}$ ), and monomers ( P ). The latter may further be present as closed hairpins or single stranded species. Rather than resolving this mixture, we used the fluorescence signal to calculate the total concentration of monomer or dimers predator species in the mixture $P_t=P+P: P$. To do this we assumed that the fluorescence signal F was dominated by the fluorescence contribution of P dimers (single stranded DNA produces less that $10 \%$ of the double stranded DNA EvaGreen fluorescence [24] and monomers in the hairpin form have only a short and unstable double strand part, which is still not expected to produce a large fluorescence signal compared to that of dimers). We could then calculate the total concentration $P_t$ at each time point from the relation expressing the fluorescence F as a function of only $\mathrm{P}: \mathrm{P}$ dimer concentration, and the expression of the thermodynamic equilibrium between monomers and dimers.

Figure S7: Kinetic of prey degradation by ttRecJ .
Left: $\mathrm{N}_1$; right: $\mathrm{N}_2$. Dots are the experimental data and solid lines are the best fits to the Michaelis-Menten equation (2).

$$
\begin{gathered}
F=\alpha \cdot P: P \\
P: P=K_a \cdot P^2
\end{gathered}
$$
giving :
$$
P_t(t)=f(t)+\sqrt{\frac{f(t)}{K_a}}, \text { with } f(t)=\frac{F(t)}{F_{t=0}} P: P_{t=0}
$$

Here, $K_a$ is the thermodynamic constant for the equilibrium between monomers and dimers and $\alpha$ the fluorescence coefficient, in arbitrary units.

Because we use a buffer containing EvaGreen and ETSSB, expected to have respectively a duplex stabilizing[24] and a duplex destabilizing [9] effect, and also because of the secondary structure on P , the value of $K_a$ was not computed from nearest neighbor model, but directly extracted form the analysis of melting curve of $\mathrm{P}_1$ and $\mathrm{P}_2$ at $1.3 \mu \mathrm{M}$ in the presence of EvaGreen and ETSSB. We found $K_a=1.4 n M^{-1}$ for $\mathrm{P}_1$ and $K_a=0.19 n M^{-1}$ for $\mathrm{P}_2$ at $46.5^{\circ} \mathrm{C}$. For reference, the computed value with the nearest-neighbor approximations were calculated to be $0.039 n M^{-1}$ for $\mathrm{P}_2$ and $0.11 n M^{-1}$ for $\mathrm{P}_1$.

We thus performed experiments in which the digestion of $\mathrm{P}_1$ or $\mathrm{P}_2$ (initially 400 nM ) by 15 nM of ttRecJ was monitored via the evolution of the fluorescence signal. We used this time series to calculate $P_t(t)$ from equation (12), and then calculated the apparent Michaelis-Menten parameters for the exonuclease ttRecJ with respect to $\mathrm{P}_{\mathrm{t}}$ concentration (Fig. S8). Since ttRecJ is reported as single-strand specific exonuclease ${ }^4$ this approximation is not absolutely correct. Note however, that, by definition, the two-species general model (developed in Section S3) does not distinguish between predators monomers and dimers, and therefore this approximation remains the most consistent choice available, for the purpose of providing numerical parameters for the model.

${ }^4$ However the activity of mesophilic RecJ has been reported to be strongly modified and enhanced by the presence of mesophilic SSB proteins [25], suggesting a yet undefined interaction between the two proteins.

Figure S8: Kinetic of predator degradation by ttRecJ .
Left: $\mathrm{P}_1$; right: $\mathrm{P}_2$. The concentration $P_t$ and the corresponding reaction speed (dots) were extracted from the analysis of the fluorescence evolutions as described and fitted with the Michaelis-Menten equation (solid line).

\begin{table}
\captionsetup{labelformat=empty}
\caption{Table S4: Degradation kinetic parameters for $\mathrm{N}_1, \mathrm{~N}_2, \mathrm{P}_1$ and $\mathrm{P}_2$}
\begin{tabular}{|c|c|c|c|c|}
\hline & $\mathrm{N}_1$ & $\mathrm{~N}_2$ & $\mathrm{P}_1$ & $\mathrm{P}_2$ \\
\hline \hline$V_m\left(n M . \mathrm{min}^{-1}\right)$ & 274 & 340 & 2.4 & 6.2 \\
\hline$k_{\text {cat }}\left(\mathrm{min}^{-1}\right)$ & 33 & 41 & 0.16 & 0.41 \\
\hline$K_m(n M)$ & 1600 & 2900 & 34 & 74 \\
\hline$k_{1 \text { st }}=k_{\text {cat }} / K_m\left(n M^{-1} . \mathrm{min}^{-1}\right)$ & $2.1 \times 10^{-2}$ & $1.4 \times 10^{-2}$ & $4.7 \times 10^{-3}$ & $5.6 \times 10^{-3}$ \\
\hline
\end{tabular}
\end{table}

Predation:
This step is studied by incubating an initial concentrations $N_{1, t=0}=150 n M$ or $N_{2, t=0}=100 n M$ of prey with various concentrations of polymerase. We run these reactions in the usual buffer, containing all enzymes except the exonuclease and at the same temperature ( $46.5^{\circ} \mathrm{C}$ ). The single reaction occurring in this case is:
$$
N+P \rightarrow 2 P
$$

Since EvaGreen-generated fluorescence of P is higher than that of N , we observe an increase in the fluorescence signal. The predator concentration at each time point is calculated from the value of the fluorescence increase by comparing it with a standard curve, itself build from measurement of various known mixtures of N and P , each corresponding to a point on the trajectory followed by the predation reaction in concentration space, (i.e. satisfying the relation $N+P=$ cste $=N_{t=0}$ ). This yielded the time plots shown in Fig. S9.

The kinetic equation corresponding to this reaction can be written (see Section S3-Predation):
$$
\frac{d P}{d t}=k_2 \cdot \mathrm{pol} \cdot P \cdot N
$$

Noting that $N=N_{t=0}-P$, this system is readily integrated to yield the time evolution of $P$ :
$$
P(t)=\frac{P_{t=0} \cdot N_{t=0} \cdot e^{N_{t=0} \cdot k_2 \cdot \mathrm{pol} \cdot t}}{P_{t=0} \cdot e^{N_{t=0} \cdot k_2 \cdot \mathrm{pol} \cdot t}-P_{t=0}+N_{t=0}}
$$

Therefore, the experimental records were fitted to this equation. The optimized values of the product $k_2 . p o l$ were then plotted as a function of the known concentration of polymerase and a linear regression yielded the value $k_{2,1}=4.8 \times 10^{-3} n M^{-2} \cdot \min ^{-1}$ (for reaction $\mathrm{N}_1+\mathrm{P}_1 \rightarrow 2 \mathrm{P}_1$ ). The same procedure was repeated for $\mathrm{N}_2$, giving $k_{2,2}=2.1 \times 10^{-3} n M^{-2} ._{\text {min }^{-1}}$ (Fig. S9).

Figure S9: Predator growth kinetics.
Left $\mathrm{P}_1$; Right $\mathrm{P}_2$. Disks are the experimental results and solid lines the best fits for each concentration of polymerase. Prey ( 150 nM for $\mathrm{P}_1$ and 100 nM for $\mathrm{P}_2$ ) and an initial small concentration of predators were incubated at $46.5^{\circ} \mathrm{C}$ in the presence of various concentrations of Bst polymerase, and the fluorescence signal recorded. Predator concentrations were calculated from the fluorescence evolution using a calibration curve as described and fitted to equation (13). The inset shows the value of the resulting kinetic rate as a function of polymerase concentration. The slope of the linear fit is $k_2$.

Prey amplification
Because the G template can form various duplex or partial duplex species in the course of prey amplification, (reaction 1 to 5 in Fig. S1), and also because this template is present in limited quantities in the reaction mix, we cannot expect the fluorescence signals (either EvaGreen fluorescence or the shift in template bound dye intensity) to follow linearly the prey concentration -at least not over a large range of prey concentrations. To obtain quantitative information about the prey amplification, we sampled the reaction mix over time and directly remeasured the prey concentrations in these aliquots using a previously reported procedure [10]. Briefly, it involved the following steps:
$\triangleright$ We initiated the prey amplification reaction in the presence of the dye-labeled G template ( 200 nM ), polymerase and nicking enzyme (standard concentrations), but in the absence ttRecJ. Initial conditions contained a small amount of prey ( 5 nM ) but no predators.
$\triangleright$ We incubated the reaction at $46.5^{\circ} \mathrm{C}$, then withdrew aliquots from the reaction at various time points and immediately quenched them by diluting 5 times in hot water and heating to $95^{\circ} \mathrm{C}$ for 5 minutes.
$\triangleright$ We then measured the prey concentration in these aliquots by real-time isothermal DNA amplification using a specifically designed template and comparing to a calibration curve built from known concentration of $\mathrm{N}_1$ and $\mathrm{N}_2$ (bearing a $5^{\prime}$ phosphate). The reaction conditions for this measurement are: [Template $\left(\mathrm{MesN}_1\right.$ for $\mathrm{N}_1$ and $\mathrm{MesN}_2$ for $\left.\left.\mathrm{N}_2\right)\right]=30 \mathrm{nM},[$ Nt.bstNBI $]=0,1$ unit $/ \mu \mathrm{L}$, pol $=16$ unit $/ \mathrm{mL}$, $T=52^{\circ} \mathrm{C}$ for $\mathrm{N}_1$ measurements and $50^{\circ} \mathrm{C}$ for $\mathrm{N}_2$.
$\triangleright$ These experiments provided direct (but time-lapsed) concentration measurements, shown in Fig. S10 together with the evolution of fluorescence in both green and yellow (red) channel for $\mathrm{PP}_1\left(\mathrm{PP}_2\right)$. Note that in this assay (i.e. in the absence of exonuclease), due to the exponential nature of the autocatalytic reaction, very high concentrations of prey are attained relatively quickly. As a consequence, predators can spontaneously appear by extension of prey dimers. They would then propagate very quickly by feeding on the high concentration of prey and completely disrupt the measurement ${ }^5$. For this reason, this experiment cannot be continued long enough to allow a precise quantification of the saturation parameter $b$. Hence $b$ is obtained with a relatively large uncertainty and its numerical value has to be treated with caution.
Integrating prey growth expression (given in Section S3, eq. (1)) between $\left(t=0, N_{t=0}\right)$ and $(t, N)$, we find:
$$
t=\frac{1}{k_1 \cdot p o l}\left[\frac{1}{G} \ln \left(\frac{N}{N_{t=0}}\right)+b\left(N-N_{t=0}\right)\right]
$$

Fitting this expression to the curves in Fig. S10 (we let $N_{t=0}$ free to avoid problems with the exact timing of reaction initiation, during mixing and heating to $46.5^{\circ} \mathrm{C}$ ) gives the values of $k_1$ and $b$ (see Table S5).

${ }^5$ The leak reaction mentioned here is unlikely to have a strong influence on the dynamic of the full oscillating PP system, because i) prey accumulate much less in the presence of exonuclease (at $150 \mathrm{nM} \mathrm{G}_1$ in the conditions of Fig. 2e, $\mathrm{N}_1$ reaches a calculated steady-state around 80 nM , if predators are absent) ii) predators will also tend to decrease the prey concentration and iii) prey dimers have only a 6 base duplex part and hence should provide a very poor substrate for the polymerase to start extension. Still we have noticed that full systems initiated in the absence of predators would eventually start to oscillate.


Figure S10: Prey amplification kinetic.
Left: $\mathrm{N}_1$; Right: $\mathrm{N}_2$. Top: prey concentrations measured from the aliquots sampled during the course of the amplification reaction. For comparison they are plotted together with the EvaGreen signal produced by the preys (green) and prey-specific template-bound fluorescence signal (orange or red) measured continuously for the same reaction. Bottom: time versus concentration measurements are fitted using equation (14).


Optimizing the parameter set
With the 'raw' set of parameters obtained above for $\mathrm{PP}_1$ (Table S5), the dynamic behavior predicted by the mathematical equations (eqs. 3 and 4) is shown in Fig. S11b. Without any fitting, the model predicts the emergence of stable oscillations above a given template concentration. These stable oscillations last for a range of $G_1$ above which damped oscillations are observed before the system settle on a coexistence steady state (Fig. S11b). This is in line with the experimental trend even if, unsurprisingly considering the approximations made both in building the model and in measuring its kinetic parameters, we do not obtain a quantitative agreement concerning the template concentrations at bifurcation, nor the periods and amplitudes. However a simple fitting process allows to compensate for the possible errors in the measured kinetic values and/or the various approximations. After this process the model describe accurately the experimental data set, over the full range of concentrations of template (Fig. S11c). Concerning the parameters values, we note that it was possible to obtain this good fit without departing very far from the initial parameter set (see Table S5). The largest adjustment generally concerns the value of $b$ which, as mentioned above, was already measured with a large uncertainty.


Table S5: Measured ('raw') and optimized kinetic parameters for prey growth, predation and degradation reactions.

\begin{tabular}{|l|l|l|l|l|l|l|}
\hline \multirow{3}{*}{} & \multicolumn{2}{|c|}{'raw' parameters (from Section S4)} & \multicolumn{4}{|l|}{optimized parameters (from PP systems fitting, with 'raw' values a initial parameter set)} \\
\hline & \multicolumn{2}{|c|}{} & \multicolumn{2}{|c|}{rec $=32.5 \mathrm{nM}$} & \multicolumn{2}{|c|}{$r e c=15 n M$} \\
\hline & $\mathrm{PP}_1$ & $\mathrm{PP}_2$ & $\mathrm{PP}_1$ (Fig. 2 and S11) & $\mathrm{PP}_2$ (Fig. S13) & $\mathrm{PP}_1$ ( Fig. S14) & $\mathrm{PP}_2$ (Fig. S15) \\
\hline $k_1\left(n M^{-2} . \min ^{-1}\right)$ & $1.3 \times 10^{-3}$ & $9.0 \times 10^{-4}$ & $2.0 \times 10^{-3}$ & $8.7 \times 10^{-4}$ & $2.0 \times 10^{-3}$ & $6.0 \times 10^{-4}$ \\
\hline $b\left(n M^{-2}\right)$ & $2.4 \times 10^{-5}$ & $6.7 \times 10^{-5}$ & $4.8 \times 10^{-5}$ & $3.1 \times 10^{-5}$ & $1.2 \times 10^{-4}$ & $5.5 \times 10^{-5}$ \\
\hline $k_2\left(n M^{-2}{ }_{m i n}{ }^{-1}\right)$ & $4.8 \times 10^{-3}$ & $2.1 \times 10^{-3}$ & $3.1 \times 10^{-3}$ & $1.5 \times 10^{-3}$ & $5.5 \times 10^{-3}$ & $2.1 \times 10^{-3}$ \\
\hline $k_N\left(n M^{-1} . \min ^{-1}\right)$ & $2.1 \times 10^{-2}$ & $1.4 \times 10^{-2}$ & $2.1 \times 10^{-2}$ & $1.2 \times 10^{-2}$ & $2.1 \times 10^{-2}$ & $0.8 \times 10^{-2}$ \\
\hline $k_P\left(n M^{-1} . \min ^{-1}\right)$ & $4.7 \times 10^{-3}$ & $5.6 \times 10^{-3}$ & $4.7 \times 10^{-3}$ & $3.3 \times 10^{-3}$ & $4.7 \times 10^{-3}$ & $5.4 \times 10^{-3}$ \\
\hline $K_{m, P}(n M)$ & 34 & 74 & 34 & 50 & 34 & 74 \\
\hline
\end{tabular}

Figure S11: Parameters optimization for $\mathrm{PP}_1$ oscillations.
(A) Experimental fluorescence records for various $\mathrm{G}_1$ concentrations with standard conditions (same set of experiments as in Fig. 2 main text). (B) Model predictions with the 'raw' set of parameter (obtained from independent kinetic measurements of the reactions, as described in Section S4). For comparison with the fluorescence data in a, the computed concentrations of prey are shown as $400-P_1$. The bifurcation diagram is given below the plots. (C) Simulated evolutions after optimization of the parameter set simultaneously on all experiments shown in $a$ (from top to bottom, $G_1=60,110160210260 \mathrm{nM}$ ). The corresponding bifurcation diagram is given below the plots.


5 A second PP pair $\left(\mathrm{PP}_2\right)$
Following the steps used to design the first PP pair, we assembled a second one ( $\mathrm{PP}_2$ ) from its elementary reactions. This process is depicted in Fig. S12 with monochrome green fluorescence monitoring. After checking that the $\mathrm{PP}_2$ system is able to display oscillations we modify its template in order to obtain a red fluorescence response from the prey $\mathrm{N}_2$ (Fig. S13). The behavior of the $\mathrm{PP}_2$ system is qualitatively equivalent to that of $\mathrm{PP}_1$, but template concentration at bifurcations, periods and amplitudes are different. It is satisfying that the $\mathrm{PP}_2$ data in Fig. S13 is also correctly described with the model developed in Section S3 and the set of parameters measured for $\mathrm{PP}_2$ in section S4. After a reasonnable optimization of these parameters, using a similar process as for $\mathrm{PP}_1$, a very good fit is obtained as well (see Table S5 for the parameter values before and after optimization).

Figure S12: Experimental assembly of a second predator-prey pair ( $\mathrm{PP}_2$ ).
The formal reaction set is the same as for $\mathrm{PP}_1$ (i.e. same reaction network as in Fig. S1) and experiments are performed in the same conditions (temperature, buffer and enzyme concentrations) than for $\mathrm{PP}_1$ (Fig. 1 main text). The green fluorescence signal monitored here is produced by the intercalating dye EvaGreen and reports on the concentration of $\mathrm{P}_2$ and $\mathrm{N}_2$ (nonspecifically). (A) Templated prey amplification ( $G_2=200 \mathrm{nM}$ ). The leveling-off is due to the saturation of the fluorescent reporting (i.e. all templates in double strand form). (B) Predation, the replication of palindromic predators $\mathrm{P}_2$, using and consuming $\mathrm{N}_2$ as primers (initially $150 n M$ ). c, Decay. Both $\mathrm{N}_2$ and $\mathrm{P}_2$ are digested by ttRecJ exonuclease ( $8 n M$ ), albeit at different rates. (D) Combining predation and decay ( $[t t R e c J]=16 n M$ ) produces a pulse of $\mathrm{P}_2$ at the condition that $N_{2, t=0}$ is large enough ( $P_{2, t=0}=1 n M$ in all 3 experiments shown). (E) In the presence of template $\mathrm{G}_2(140 \mathrm{nM}$, standard conditions) the prey amplification mechanism resumes after the predator population has shrunk, and oscillations are obtained. (F) Sequence and modifications of the $\mathrm{G}_2$ template.

Figure S13: Two colors $\mathrm{PP}_2$ system and model.
(A) A Dy681 fluorescent modification on the 3' end of $\mathrm{G}_2$ allows the real time monitoring of the $\mathrm{N}_2$ population through N-quenching[12] in the red channel. (B) Experimental (standard conditions, $G_2$ as indicated) and simulated time-course of green and red fluorescence signals for various $\mathrm{G}_2$ concentrations (offsetted for clarity). Dots show the prey (red) and predator (green, corrected from the prey contribution) fluorescence experimental records. The continuous line is the model prediction, using the same set of optimized parameters for all four experiments (given in Table S5 for rec $=32.5 \mathrm{nM}$ ). ( $\mathbf{C}$ ) Experimental 2D trajectories in the fluorescence space. (D) Computed trajectories in $P_2-N_2$ concentration space.


6 Multispecies systems
Vandermeer [26,27] has considered coupled two PP pair systems were the coupling arises either form the fact that each predator can consume both kind of prey -albeit with some amount of preference- or by resource competition between the two preys. He could numerically predict the apparition of synchronized oscillations, period splitting, but also various forms of chaotic trajectories. Beninca et al. [28] also considered theoretically a system with two PP pairs connected with various modes of coupling. He showed that inter-pair competition could lead to synchronization with various phase shifts between the two oscillations. This two competing PP mechanism was shown to be one of the driving force in an experimental multispecies food web showing chaotic oscillations. The relation between theoretical ecosystems stability and complexity, with respect to the nature of the interactions between species (PP, competition or symbiosis) has been recently reexplored [29].

In the batch experiment presented here, as noted in the main text, permanent regimes are forbidden as the chemical potential of the system continuously decrease as the reaction progresses. It is therefore difficult to assign a clear-cut classification to the observed transient trajectories. We note however that chaos has been theoretically studied in closed chemical systems [30]. The added complexity, when comparing with open systems in permanent regimes, is that as the reaction evolves and experimental conditions change by consumption of reactants and accumulation of waste products it is possible that the system crosses some boundaries in the corresponding pseudopermanent bifurcation diagram. For example, the system might temporarily crosses chaotic zones, before going back to a simpler orbit. Such a case would still display sensitivity to the initial conditions, because these small differences would subsist until the moment when the system enters the chaotic zone, then be amplified. Note also that it has been remarked that permanent regimes are excluded from ecological systems as well, because they are generally subjected to long term trends [28] (for example due to changes in the environment or through Darwinian evolution [31, 32]).

In order to avoid a too high total concentration of templates in the reaction mix (potentially leading to parasitic reactions), we performed the multispecies experiments in conditions containing approximately half concentrations of each template $\mathrm{G}_1$ and $\mathrm{G}_2$. To preserve the oscillations in each system we simultaneously decreased the concentration of ttRecJ (to 15 nM , i.e. also approximatively half of its previous value, as suggested by the model). Oscillations of both individual systems in these new conditions are shown in Fig. S14 and S15. These experiments were used to optimize the set of parameters. Most values were again close to the initial 'raw' set; only $b$ changed significantly relative to both its 'raw' value and the one found in the previous optimization (at higher ttRecJ concentration, see Table S5). This suggests that, for unexplored mechanistic reasons, the parameter $b$ should be considered as a function of ttRecJ enzyme concentration, rather that a true constant.

Figure S14: $\mathrm{PP}_1$ oscillations at low ttRecJ.
$[t t R e c J]=15 n M$, other conditions as usual. (A) Time series for various concentrations of the template $\mathrm{G}_2$. The dots shows the experimental data. The line is the best simultaneous fit to these four experiments (i.e. using a single set of parameters). (B) Experimental trajectories in the fluorescence space. (C) Calculated trajectories in predator-prey concentration space. (D) Calculated bifurcation diagram using the parameter value from the fit. The red dashed line represent the $\lambda$ value obtained from the optimization process.

Figure S15: $\mathrm{PP}_2$ oscillations at low ttRecJ.
$[t t R e c J]=15 n M$, other conditions as usual. (A) Time series for various concentration of the template $\mathrm{G}_2$. The dots shows the experimental data. The line is the best simultaneous fit to these four experiments (i.e. using a single set of parameters). (B) Experimental trajectories in the fluorescence space. (C) Calculated trajectories in predator-prey space. (D) Calculated bifurcation diagram using the parameter value from the fit. The red dashed line represent the $\lambda$ value obtained from the optimization process.

Figure S16: Competition for catalytic resources between $\mathrm{PP}_1$ and $\mathrm{PP}_2$.
The reaction network is the same as in Fig. 3 of the main text and shown schematically in the upper left corner. The time plots of fluorescence records in yellow, red and green channels, corresponding respectively to $\mathrm{N}_1, \mathrm{~N}_2$ and -after subtraction of the preys' contribution- $\mathrm{P}_1+\mathrm{P}_2$. Traces are offseted for clarity. The concentrations of $\mathrm{G}_1$ (from 0 to 90 nM ) and $\mathrm{G}_2$ (from 0 to 160 nM ) in each experiment is given by the position of the plot in the matrix.

Figure S17: Competition for catalytic resources between $\mathrm{PP}_1$ and $\mathrm{PP}_2$, Fourier analysis.
Same as in Fig. S16, but the discrete Fourier transform (normalized to the intensity of the largest peak) of the yellow and red signals are shown in the corresponding color (thick lines), and compared to the situation were each pair is present alone in solution (dashed lines), with the same template concentration (i.e. for each element of the matrix, the corresponding top-row and first-column plot).

Figure S18ï¼Competition for catalytic resources between $\mathrm{PP}_1$ and $\mathrm{PP}_2$ ï¼twoï¼dimensional trajectoriesï¼ Same as in Figï¼S16ï¼but 2ï¼dimensional trajectoriesï¼plotted as $N_1-N_2$ versus $P_1+P_2$ ï¼are shownï¼

6.1 Competition

Figs. S16 to S18 show a set of experiments with two competing PP pairs, for various $G_1$ and $G_2$ concentrations. Each competitive experiment is compared to non-competitive case, i.e. the situation where only one template is present at the same concentration, shown in the top row ( $\mathrm{PP}_1$ only) and leftmost column ( $\mathrm{PP}_2$ only) of the matrix. The coupling of the two systems is obvious from the modifications of the Fourier transform of each prey signal (Fig. S17). The complexity of the behavior is better observed in the two-dimensional trajectories drawn in Fig. S18.

Mathematical model
When two PP pairs are active in the same mix, the DNA species will compete for the catalytic (enzymatic) resources. In the present mathematical model, the degradation reaction is represented as a Michaelis-Menten term with relatively small $K_m \mathrm{~s}$ (compared to the amplitude of predator oscillations (e.g. Fig. 2g of the main text or Fig. S15) and hence it is natural to include a competitive inhibition term in the corresponding denominators of the four species ordinary differential equations system ${ }^6$ :
$$
\begin{gathered}
\dot{N}_1=k_1^1 \cdot \mathrm{pol} \cdot G_1 \frac{N_1}{1+b \cdot G_1 \cdot N_1}-k_2^1 \cdot \mathrm{pol} \cdot N_1 \cdot P_1-\mathrm{rec} \cdot k_{N_1} \frac{N_1}{1+\frac{P_1}{K_{m, P_1}}+\frac{P_2}{K_{m, P_2}}} \\
\dot{P}_1=k_2^1 \cdot \mathrm{pol} \cdot N_1 \cdot P_1-\mathrm{rec} \cdot k_{P_1} \frac{P_1}{1+\frac{P_1}{K_{m, P_1}}+\frac{P_2}{K_{m, P_2}}} \\
\dot{N}_2=k_1^2 \cdot \mathrm{pol} \cdot G_2 \cdot \frac{N_2}{1+b \cdot G_2 \cdot N_2}-k_2^2 \cdot \mathrm{pol} \cdot N_2 \cdot P_2-\mathrm{rec} \cdot k_{N_2} \frac{N_2}{1+\frac{P_1}{K_{m, P_1}}+\frac{P_2}{K_{m, P_2}}} \\
\dot{P}_2=k_2^2 \cdot \mathrm{pol} \cdot N_2 \cdot P_2-\mathrm{rec} \cdot k_{P_2} \frac{P_2}{1+\frac{P_1}{K_{m, P_1}}+\frac{P_2}{K_{m, P_2}}}
\end{gathered}
$$

In fact this is just an extension of the competitive degradation process that happens within one PP pair between the predator and the prey (see section S3-Degradation).

Figure S19: Numerical integration of the four species system.
Direct numerical integration of eqs 15 to 18 required a very small integration step and yielded unrealistic values for some of the concentrations. The parameters used here come from the optimized sets for rec $=15 \mathrm{nM}$ with $G_1=50 \mathrm{nM}$ and $G_2=100 n M . N_1, N_2$ and $P_1+P_2$ are in orange, red and green respectively.

${ }^6$ In terms of Michaelis Menten kinetics, degradation Km is at 34 nM (PP1) or 74 nM (PP2), while production Km (for 100 nM template, standard conditions) are around 200 nM (for PP1) and 300 nM (PP2) (see Table S5)

Note that direct numerical simulation of such coupled systems yielded in most cases unstable results, with extremely low minimum values for one of the prey (Fig. S19). For comparison, a single molecule in a $20 \mu \mathrm{~L}$ test tube represent a concentration of $10^{-8} n M$. Hence the orange curve representing $N_1$ in Fig. S19, where concentrations goes down to $10^{-17} n M$, has no physical interpretation in our setup. Moreover stochastic behaviors are also unlikely to happen in the real system (at least for large, $\mu \mathrm{L}$, volumes) because the reaction on which prey amplification is based is reported to display some amount of leak (i.e. production of output in the absence of input[4]). To introduce such effect in the model, we have added a small constant in the numerator of the corresponding fluxes, which can be interpreted as a measure of leaky, but polymerase and template-dependent production of prey:
$$
\varphi_{N \triangleright N}=\operatorname{pol} \cdot G \frac{k_1 \cdot N+\epsilon}{1+b \cdot G \cdot N}
$$

The constant was arbitrarily set to $\epsilon=10^{-7}$ min $^{-1} n M^{-1}$. As can be seen in Fig. SS21 this value was enough to stabilize the numerical integration and avoid numerical "extinctions" of prey. Still it had only a minor impact on the numerical integration in the case of a single PP pair. Such leak reaction may experimentally correspond to de novo synthesis of preys. Another possibility is that the model does not consider the alternative chemical forms of the preys, for example un-nicked duplex template that would constitute a form of hidden prey reservoir, and tend to slow down their decay.

By numerically integrating in these conditions, we observe that the coupling at the exonuclease level is enough to drive the system toward a very complicated bifurcation landscape, as can be seen in Fig. S20 (left column). Fig. S21 shows a few representative orbits for a range of template concentrations. Chaotic behavior, synchronization and higher periodicities can be observed. Moreover some of the orbits are strongly reminiscent of the experimental results.

It is however also possible to consider that the prey growth is a saturable process as well (even if it is not chemically a pure Michaelis-Menten reaction). For the conditions of Fig. 3 in the main text, and considering i) the respective values of the saturation terms ( $K_{m, P}<1 / b . G$ in both cases) and ii) that the model predicts that prey accumulates approximatively twice less than predators, we may assume that the coupling at this stage is weaker than that of the degradation process and may be less significant with regard to the global dynamics. We have still built a model that includes this latter effect. The mathematical expression is similar to the previous one except that the growth expressions for the preys are replaced by coupled fluxes:
$$
\varphi_{N_i \triangleright N_i}=\operatorname{pol} \cdot G_i \frac{k_{1 i} \cdot N_i+\epsilon}{1+\sum b \cdot G_i \cdot N_i} \quad i=1,2
$$

Fig. S20 compares the bifurcation patterns for both hypothesis (coupling via degradation only or coupling via degradation and production) for various templates conditions. While the detailed behaviors are obviously different, the global trend, showing alternation between chaotic zones and areas of synchronized behavior, generally with higher periodicities, is maintained.

Figure S20: Bifurcations in enzymatically coupled four species models.
Left: assuming that coupling between the two PP pairs happens only through competition for ttRecJ exonuclease. Right: assuming competition occurs in both prey growth and degradation processes. These plots show numerical values obtained by searching for all local maxima and minima in 5000 minutes of the calculated $\mathrm{P}_2$ trajectory for eqs. (15) to (18). This search was started 5000 minutes after the first calculated point, in order to remove transient parts of the trajectories. Initial conditions are $N_1=N_2=P_1=P_2=10 n M$.

Figure S21: Computed trajectories for two PP pairs competing for ttRecJ.
Left column: three-dimensional trajectory in $N_1 / N_2 / P_1+P_2$ concentration space (corresponding to the 3 experimentally observable fluorescence signals); Middle column: 2D plots of $P_1+P_2$ versus $N_2-N_1$; Right column: time plots of $N_1$ (orange), $N_2$ (red) and $P_1+P_2$ (green). For clarity, and in line with the observed negative fluorescence shifts, we have plotted $-N_1$ and $-N_2$ instead of $N_1$ and $N_2$. The concentration of $G_2$ is 80 nM for all the plots. That of $G_1$ is changed for each row, as indicated. The range of behavior observed includes periodic oscillations (at low $G_1$ ), synchronization, perioddoubling and chaos. Trajectories and time plots are shown between $\mathrm{t}=7000$ and $\mathrm{t}=10000$ minutes. Initial conditions are $N_1=N_2=P_1=P_2=10 \mathrm{nM}$.

6.2 Symbiosis

Predator-Prey-mutualist
Equations (3) and (4) are readily modified to incorporate the reactions driven by the two templates $\mathrm{G}_{2 \rightarrow S y}$ and $\mathrm{G}_{S y \rightarrow 2}$. For simplicity, and in line with the experiments presented in this report, we limit ourselves to the case where both of these templates have the same concentration, noted $G_{S y}$. Noting that $G_{S y} \ll G_2$ in all cases, we also assume that $\mathrm{G}_{\mathrm{Sy}}$ don't significantly participate in the saturation of the enzymatic machinery and we can this neglect the corresponding term in the denominator of the prey production term (that is we write it as $1+b \cdot G_2 \cdot N_2$ and not as $1+b . G_2 N_2+b^{\prime} . G_{S y} . S y$ ). Finally, since Sy and $\mathrm{N}_2$ are chemically very close (same length and similar sequences), we use the same first order degradation rate for both (with the numerical value obtained for $\mathrm{N}_2$ in Section S4). With obvious notations we can then write:
$$
\begin{gathered}
\dot{N}_2=\operatorname{pol} \frac{k_1 \cdot G_2 \cdot N_2+k_{S y} \cdot G_{S y} \cdot S y}{1+b \cdot G_2 \cdot N_2}-k_2 \cdot \mathrm{pol} \cdot N_2 \cdot P_2-\mathrm{rec} \cdot k_N \frac{N_2}{1+\frac{P_2}{K_{m, P}}} \\
\dot{S y}=k_{S y} \cdot \mathrm{pol} \cdot \frac{G_{S y} \cdot N_2}{1+b \cdot G_2 \cdot N_2}-\mathrm{rec} \cdot k_N \frac{\mathrm{Sy}}{1+\frac{P_2}{K_{m, P}}} \\
\dot{P}_2=k_2 \cdot \mathrm{pol} \cdot N_2 \cdot P_2-\mathrm{rec} \cdot k_p \frac{P_2}{1+\frac{P_2}{K_{m, P}}}
\end{gathered}
$$

Fig. S22 shows the effect of this modification on the predicted dynamics: at low coupling strength (small $k_{s y} . G_{S y}$ ), Sy oscillates in phase with the prey; when the symbiotic link becomes strong enough, the oscillations dampen and the system tends to stabilize with coexistence of all species. These observations are in line with the experimental results and also agree with a previous theoretical study of a Predator-Prey-Mutualist system[33]. From the plots in Fig. S22, we obtain an estimated value of $k_{S y}=10^{-3} n M^{-2} \cdot \min ^{-1}$ (with $k_1=.0006 n M^{-2} \cdot \min ^{-1}$ for $\mathrm{PP}_2$ low rec). The maximum $G_{S y}$ value allowing stable limit cycles is numerically found $5 n M$. This value should be compared to the concentration of $\mathrm{G}_2$ in this experiment, $G_2=120 \mathrm{nM}$.

Figure S22ï¼Predatorï¼preyï¼mutualist experiment and modelï¼
ï¼Aï¼Experimental dataï¼same as in Figï¼ 4 of the main textï¼ï¼ï¼Bï¼Simulated fluroescent trajectories and 2ï¼dimensionnal trajectories in concentration spaceï¼eqs 19 to 21ï¼ï¼The parameters for the model are those of Table S5 for $\mathrm{PP}_2$ low recï¼with $k_{s y}=10^{-3} n M^{-2} . \min ^{-1}$.

Two-PP system with mutualistic preys
Following a similar reasoning, eqs. (15) to (18) are modified to include the effect of the two symbiotic templates $\mathrm{G}_{1 \rightarrow 2}$ and $\mathrm{G}_{2 \rightarrow 1}$ (present at identical concentrations). The prey growth terms are replaced by the fluxes:
$$
\varphi_{N_i \triangleright N_i}=p o l \frac{G_i\left(k_{1 i} \cdot N_i+\epsilon\right)+\sigma \cdot N_{j \neq i}}{1+b \cdot G_i \cdot N_i} \quad i=j=1,2
$$
where $\sigma$ is a measure of the activity of templates $\mathrm{G}_{1 \rightarrow 2}$ and $\mathrm{G}_{2 \rightarrow 1}$, assumed to be equal. This yield the plots in Fig. S23. We observe that very small values of $\sigma$ (compared to $k_1 \cdot G_1=0.09 \mathrm{nM}^{-1} \min ^{-1}$ and $k_2 \cdot G_2= 0.06 \mathrm{nM}^{-1} \min ^{-1}$ ) have a strong impact on the dynamic. This is in line with the experimental observations where very small concentrations of $\mathrm{G}_{1 \rightarrow 2}$ and $\mathrm{G}_{2 \rightarrow 1}$, at the level of several nanomolars are enough to completely modify the behavior (Fig. 4 of the main text). As $\sigma$ increases, we observe a gradual overall synchronization of the two pairs but, in contrast with the previous three-species system, the oscillatory behavior is maintained.

Figure S23: Predicted effect of increasing symbiotic interaction between $\mathrm{N}_1$ and $\mathrm{N}_2$ for the two PP pairs system. The model assumes saturation at the degradation level only. Left column: Trajectory in $N_1 / N_2 / P_1+P_2$ concentrations space (corresponding to the three experimentally observable fluorescence signals); Middle column: 2D plots of $P_1+P_2$ versus $N_2-N_1$; Right column: time plot of $-N_1$ (orange), $-N_2$ (red) and $P_1+P_2$ (green). The concentration of $\mathrm{G}_2$ is 100 nM and that of $\mathrm{G}_1$ is 45 nM for all the plots. $\sigma$, representing the strength of the mutualistic interaction (or the activity of $\mathrm{G}_{1 \rightarrow 2}$ and $\mathrm{G}_{2 \rightarrow 1}$ templates, given in $\mathrm{nM}^{-1} \min ^{-1}$ ), is increased as one goes down the rows, as indicated. Trajectories and time plots are shown between $t=7000$ and 10000 minutes.

7 Legends for supplementary movies
$\triangleright$ Movie S1: 3-dimensional animated version of the plot in Fig. 4c (top plot, i.e. competition between two PP pairs with $\left.\mathrm{G}_1=50 n M ; \mathrm{G}_2=100 n M\right)$, in the fluorescence space (arbitrary units). The time has been accelerated 2700 x .
$\triangleright$ Movie S2: 3-dimensional animated version of the plot in Fig. 4c (bottom plot, i.e. competition between two PP pairs with $\left.\mathrm{G}_1=60 n M ; \mathrm{G}_2=110 n M\right)$, in the fluorescence space (arbitrary units). The time has been accelerated 2700x.
- Movie S3: 3-dimensional animated version of the plot in Fig. 4f, in the fluorescence space (arbitrary units), showing the divergence of trajectories for multispecies experiments initiated with slightly different conditions. Black: $N_1=N_2=P_1=P_2=10 n M$; blue: $N_1=12 n M, N_2=P_1=P_2=10 n M$. Red: $N_1=P_1=P_2= 10 n M, N_2=12 n M$. The time has been accelerated 2700x.
